---
title: "proteomics.bayesPrism_deconv"
author: "FloWuenne"
date: "2023-06-14"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r}
library(data.table)
library(BayesPrism)
library(Seurat)
library(pheatmap)
library(tidyverse)
```


# Introduction

Here we will use BayesPrism to attempt to deconvolute the protomics protein matrix using snRNA-seq data from [Calcagno et al. 2022](https://www.nature.com/articles/s44161-022-00160-3).

# Load and format data 

```{r}
## Import imputed protein table (our "bulk" data)
imp_prot <- fread(file = "./output/proteomics.filt_imputed_proteins.tsv")
proteins <- imp_prot$Genes
imp_prot_mat <- imp_prot %>%
  select(-c(Protein_Ids,imputed_values,Genes))
samples <- colnames(imp_prot_mat)
imp_prot_mat <- t(imp_prot_mat)
imp_prot_mat <- as.matrix(imp_prot_mat)
imp_prot_mat <- round(2^imp_prot_mat,0)
rownames(imp_prot_mat) <- samples
colnames(imp_prot_mat) <- proteins
```

```{r}
calcagno_et_al_seurat <- readRDS("./data/140623.calcagno_et_al.seurat_object.rds")
```

```{r}
## Filter out lowly abundant cell-types
calcagno_et_al_seurat <- subset(calcagno_et_al_seurat,final_cluster != "Sigf")

## Create BayesPrism input data
sc.dat <- as.matrix(t(calcagno_et_al_seurat@assays$RNA@counts))
cell.type.labels <- as.character(calcagno_et_al_seurat@meta.data$cell_type)
cell.state.labels <- as.character(calcagno_et_al_seurat@meta.data$final_cluster)
```


```{r}
## Plot a table showing cell-type and cell state numbers
table(cbind.data.frame(cell.state.labels, cell.state.labels))
```

```{r}
## Plot correlation of cell-types
plot.cor.phi (input=sc.dat, 
              input.labels=cell.type.labels, 
              title="cell type correlation",
              #specify pdf.prefix if need to output to pdf
              #pdf.prefix="gbm.cor.ct",
              cexRow=0.5, cexCol=0.5,
                         )
```

# Filter data

Next, we will filter the data before runnin BayesPrism.


## Filter outlier genes

```{r}
sc.stat <- plot.scRNA.outlier(
  input=sc.dat,
  cell.type.labels=cell.type.labels,
  species="mm",
  return.raw=TRUE
)
```

```{r}
head(sc.stat)
```

```{r}
sc.dat.filtered <- cleanup.genes(input=sc.dat,
                                  input.type="count.matrix",
                                  species="mm", 
                                  gene.group=c( "Rb","Mrp","other_Rb","chrM","chrX","chrY") ,
                                  exp.cells=5)
```

## Determine signature genes


```{r}
diff.exp.stat <- get.exp.stat(sc.dat=sc.dat.filtered[,colSums(sc.dat.filtered>0)>3],# filter genes to reduce memory use
                              cell.type.labels=cell.type.labels,
                              cell.state.labels=cell.state.labels,
                              psuedo.count=0.1, #a numeric value used for log2 transformation. =0.1 for 10x data, =10 for smart-seq. Default=0.1.
                              cell.count.cutoff=50, # a numeric value to exclude cell state with number of cells fewer than this value for t test. Default=50.
                              n.cores=1 #number of threads
                                          )
                                          
```


```{r}
sc.dat.filtered.sig <- select.marker (sc.dat=sc.dat.filtered,
                                         stat=diff.exp.stat,
                                         pval.max=0.01,
                                         lfc.min=0.1)
```

# Run BayesPrism

```{r}
myPrism <- new.prism(
  reference=sc.dat.filtered, 
  mixture=imp_prot_mat,
  input.type="count.matrix", 
  cell.type.labels = cell.type.labels, 
  cell.state.labels = cell.state.labels,
  key=NULL,
  outlier.cut=0.01,
  outlier.fraction=0.1,
)
```

```{r, message = FALSE,warning = FALSE}
bp.res <- run.prism(prism = myPrism, n.cores=50)
```

## Get cell type fractions

```{r}
theta <- get.fraction (bp=bp.res,
            which.theta="final",
            state.or.type="type")
```

```{r}
pheatmap(theta)
```


## Extract posterior mean of cell-type specific gene expression

```{r}
## Endocardial cells
endocard <- get.exp (bp=bp.res,
                          state.or.type="type",
                          cell.name="Endocardial_cells")
endocard_df <- t(endocard)

## Macrophages
macros <- get.exp (bp=bp.res,
                          state.or.type="type",
                          cell.name="Macrophages")
macros_df <- t(endocard)

## Skap1
skap1 <- get.exp (bp=bp.res,
                          state.or.type="type",
                          cell.name="Skap1")
skap1 <- t(endocard)
```

## Save Prism object and results

```{r}
bp.res <- saveRDS(bp.res,
                  file = "./data/proteomics.bayesprism.bp_res.rds")
```
