<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Florian Wuennemann" />


<title>Figure 4</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mi_spatialomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="data_processing.html">Data processing</a>
</li>
<li>
  <a href="data_analysis.html">Data analysis</a>
</li>
<li>
  <a href="figures.html">Figures</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Figure 4</h1>
<h4 class="author">Florian Wuennemann</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-03-21
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>mi_spatialomics/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230612code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230612)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230612code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230612)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomFloWuennemispatialomicstreee6213a5ae03dcd38ce227600f6ad3c4c9a7c5fc0targetblanke6213a5a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/FloWuenne/mi_spatialomics/tree/e6213a5ae03dcd38ce227600f6ad3c4c9a7c5fc0" target="_blank">e6213a5</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomFloWuennemispatialomicstreee6213a5ae03dcd38ce227600f6ad3c4c9a7c5fc0targetblanke6213a5a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/FloWuenne/mi_spatialomics/tree/e6213a5ae03dcd38ce227600f6ad3c4c9a7c5fc0" target="_blank">e6213a5</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/deprecated/.DS_Store
    Ignored:    analysis/molecular_cartography_python/.DS_Store
    Ignored:    analysis/seqIF_python/.DS_Store
    Ignored:    analysis/seqIF_python/pixie/.DS_Store
    Ignored:    analysis/seqIF_python/pixie/cell_clustering/
    Ignored:    annotations/.DS_Store
    Ignored:    annotations/SeqIF/.DS_Store
    Ignored:    annotations/molkart/.DS_Store
    Ignored:    annotations/molkart/Figure1_regions/.DS_Store
    Ignored:    annotations/molkart/Supplementary_Figure4_regions/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/140623.calcagno_et_al.seurat_object.rds
    Ignored:    data/Calcagno2022_int_logNorm_annot.h5Seurat
    Ignored:    data/IC_03_IF_CCR2_CD68 cell numbers.xlsx
    Ignored:    data/Traditional_IF_absolute_cell_counts.csv
    Ignored:    data/Traditional_IF_relative_cell_counts.csv
    Ignored:    data/pixie.cell_table_size_normalized_cell_labels.csv
    Ignored:    data/results_cts_100.sqm
    Ignored:    data/seqIF_regions_annotations/
    Ignored:    data/seurat/
    Ignored:    output/.DS_Store
    Ignored:    output/mol_cart.harmony_object.h5Seurat
    Ignored:    output/molkart/
    Ignored:    output/proteomics/
    Ignored:    output/results_cts.lowres.125.sqm
    Ignored:    output/seqIF/
    Ignored:    pipeline_configs/.DS_Store
    Ignored:    plots/
    Ignored:    references/.DS_Store
    Ignored:    renv/.DS_Store
    Ignored:    renv/library/
    Ignored:    renv/staging/

Untracked files:
    Untracked:  analysis/deprecated/figures.supplementary_figureX.Rmd
    Untracked:  analysis/deprecated/figures.supplementary_figure_X.MistyR.Rmd

Unstaged changes:
    Deleted:    analysis/figures.supplementary_figureX.Rmd
    Deleted:    analysis/figures.supplementary_figure_X.MistyR.Rmd
    Deleted:    analysis/figures.supplementary_figure_X.proteomics_qc.Rmd
    Deleted:    figures/Figure_5.eps
    Deleted:    figures/Figure_5.pdf
    Deleted:    figures/Figure_5.png
    Deleted:    figures/Figure_5.svg
    Deleted:    figures/Supplementary_Figure_1_Molecular_Cartography_ROIs.png
    Deleted:    figures/Supplementary_figure_5.segmentation_metrics.poster.eps
    Modified:   figures/Supplementary_figure_X.proteomics.eps
    Modified:   figures/Supplementary_figure_X.proteomics.png
    Deleted:    results_cts.lowres.125.sqm

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/figures.Figure4.DVP.Rmd</code>)
and HTML (<code>docs/figures.Figure4.DVP.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/e6213a5ae03dcd38ce227600f6ad3c4c9a7c5fc0/analysis/figures.Figure4.DVP.Rmd" target="_blank">e6213a5</a>
</td>
<td>
FloWuenne
</td>
<td>
2024-03-21
</td>
<td>
Publish all updated files.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/56559c732914d3853c5fb73bef8b60761246d181/analysis/figures.Figure4.DVP.Rmd" target="_blank">56559c7</a>
</td>
<td>
FloWuenne
</td>
<td>
2024-03-21
</td>
<td>
Cleaned up repository.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/c1350a5a86345a77f0d220890b8c0ade0a0cd56e/analysis/figures.Figure4.DVP.Rmd" target="_blank">c1350a5</a>
</td>
<td>
FloWuenne
</td>
<td>
2024-03-07
</td>
<td>
Updated proteomic analsysi and several plots for molkart.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/f35df9defd28d524467c60e2b4e3c7c02934dcb2/analysis/figures.Figure4.DVP.Rmd" target="_blank">f35df9d</a>
</td>
<td>
FloWuenne
</td>
<td>
2024-03-02
</td>
<td>
Updated figures for proteomic results.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/8a065e904c321dbcdd60ccbfb9f9983c6a5d1204/analysis/figures.Figure4.DVP.Rmd" target="_blank">8a065e9</a>
</td>
<td>
FloWuenne
</td>
<td>
2024-03-01
</td>
<td>
Updated proteomic results figures.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/b06dcd3db178ee99174eccc48ef69c5e412d3b38/analysis/figures.Figure4.DVP.Rmd" target="_blank">b06dcd3</a>
</td>
<td>
FloWuenne
</td>
<td>
2024-02-25
</td>
<td>
Updated Figure 1,4 and S4 and 5 code.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/0941f3fdb2410aa86a6ddeb2b1ef995d09b4b2ab/analysis/figures.Figure4.DVP.Rmd" target="_blank">0941f3f</a>
</td>
<td>
FloWuenne
</td>
<td>
2024-02-15
</td>
<td>
Updated Figure 4.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-data" class="section level1">
<h1>Load data</h1>
<pre class="r"><code>pca_res &lt;- readRDS(&quot;./output/proteomics/proteomics.pca_res.rds&quot;)
vsn_mat &lt;- fread(&quot;./output/proteomics/proteomics.vsn_norm_proteins.tsv&quot;)
limma_res &lt;- fread(&quot;./output/proteomics/proteomics.limma.full_statistics.tsv&quot;)
mi_pathways &lt;- fread(&quot;./output/proteomics/proteomics.pathway_results.MIiz_MIremote.tsv&quot;)</code></pre>
</div>
<div id="subfigure-a" class="section level1">
<h1>Subfigure A</h1>
</div>
<div id="subfigure-b---principal-component-analysis"
class="section level1">
<h1>Subfigure B - Principal component analysis</h1>
<pre class="r"><code>pcs &lt;- as.data.frame(pca_res$x)
pcs$sample &lt;- colnames(vsn_mat[,1:11])
pcs &lt;- pcs %&gt;%
  mutate(&quot;group&quot; = if_else(grepl(&quot;control&quot;,sample),&quot;control&quot;,
                          if_else(grepl(&quot;MI_IZ&quot;,sample),&quot;MI_IZ&quot;,&quot;MI_remote&quot;))
         )

## Set order of groups
pcs$group &lt;- factor(pcs$group,
                    levels = c(&quot;control&quot;,&quot;MI_remote&quot;,&quot;MI_IZ&quot;))

## Plot PCs
pca_plot &lt;- ggplot(pcs,aes(PC1,PC2)) +
  geom_point(size = 5,pch = 21,color = &quot;black&quot;, aes(fill = group)) +
  ggforce::geom_mark_ellipse(color = &quot;white&quot;,aes(fill = group)) +
  expand_limits(y = c(-50, 40),
                x = c(-40,80)) +
  scale_fill_manual(values = proteome_palette,
                    labels = c(&quot;Control&quot;,&quot;MI_remote&quot;,&quot;MI_IZ&quot;)) +
  labs(color = &quot;Group&quot;) +
  guides(fill=guide_legend(title=&quot;Group&quot;)) +
  theme(legend.position = &quot;none&quot;)


pca_plot</code></pre>
<p><img src="figure/figures.Figure4.DVP.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>save_plot(filename = &quot;./plots/Figure_5.pca_plot.pdf&quot;,
          plot = pca_plot,
          base_asp = 1,
          base_height = 4)</code></pre>
</div>
<div id="subfigure-c---volcano-plot-remote-vs-control"
class="section level1">
<h1>Subfigure C - Volcano plot: Remote vs control</h1>
<pre class="r"><code>limma_mi_remote &lt;- subset(limma_res,analysis == &quot;MI_IZ_vs_MI_remote&quot;)
limma_remote_control &lt;- subset(limma_res,analysis == &quot;MI_remote_vs_control&quot;)
limma_mi_control &lt;- subset(limma_res,analysis == &quot;MI_IZ_vs_control&quot;)

## Which proteins are differentially expressed in MI vs remote but not in MI vs remote but not in remote vs control?
iz_uniq &lt;- setdiff(subset(limma_mi_remote,adj.P.Val &lt; 0.05)$gene,subset(limma_remote_control,adj.P.Val &lt; 0.05)$gene)
limma_mi_remote_uniq &lt;- subset(limma_mi_remote, gene %in% iz_uniq) %&gt;%
  subset(adj.P.Val &lt; 0.05) %&gt;%
  arrange(desc(logFC))

## Get proteins from Coagulation pathway from pathway analysis results to highlight on volcano plot
mh_gsea_net &lt;- readRDS(&quot;./references/mh.all.v2023.1.Mm.symbols.sets.rds&quot;)

pathway &lt;- &#39;HALLMARK_COAGULATION&#39;

df &lt;- mh_gsea_net %&gt;%
  filter(source == pathway) %&gt;%
  arrange(target)

path_de_inter &lt;- sort(intersect(limma_mi_remote$gene,df$target))</code></pre>
<pre class="r"><code># top_10_proteins &lt;- limma_mi_remote %&gt;%
#   arrange(desc(logFC)) %&gt;%
#   top_n(wt = logFC, 10)
# top_10_proteins &lt;- top_10_proteins$gene
# bottom_10_proteins &lt;- limma_mi_remote %&gt;% arrange(desc(logFC))
# bottom_10_proteins &lt;- tail(bottom_10_proteins,n=10)

manual_labeled_proteins &lt;- c(&quot;Thbd&quot;,&quot;Vwf&quot;,&quot;Coro1a&quot;,&quot;Thbs1&quot;)

limma_mi_remote &lt;- limma_mi_remote %&gt;%
  # mutate(&quot;label_protein&quot; = if_else(gene %in% path_de_inter &amp; adj.P.Val &lt; 0.05 &amp; (logFC &gt; 1.25 | logFC &lt; 0), gene, &quot;&quot;))
  mutate(&quot;label_protein&quot; = if_else(gene %in% manual_labeled_proteins,gene,&quot;&quot;))

limma_mi_remote$label_protein &lt;- gsub(&quot;Vwf&quot;,&quot;vWF&quot;,limma_mi_remote$label_protein)

volc_limma_IZ_remote &lt;- plot_pretty_volcano(limma_mi_remote, 
                                      pt_size = 2,
                                      plot_title = &quot;&quot;,
                                      sig_thresh = 0.05,
                                      col_pos_logFC = proteome_palette[[&#39;MI_IZ&#39;]],
                                      col_neg_logFC = proteome_palette[[&#39;MI_remote&#39;]]) +
    # geom_point(data = subset(limma_mi_remote, gene %in% path_de_inter),pch = 21, color = &quot;black&quot;, size = 4) +
  geom_label_repel(box.padding = 0.5, max.overlaps = Inf) +
  geom_vline(xintercept= 0 , linetype = 2)

## Interactive plotly plot to view genes on points
# plot_ly(data = limma_mi_remote, x = ~logFC, y = ~-log10(adj.P.Val),
#         text = ~paste(&quot;Gene: &quot;, gene))

save_plot(filename = &quot;./plots/Figure_5.volcano_plot.pdf&quot;,
          plot = volc_limma_IZ_remote,
          base_asp = 1.3,
          base_height = 3.25)</code></pre>
<pre><code>Warning: Removed 58 rows containing missing values (`geom_label_repel()`).</code></pre>
<pre class="r"><code>## Volcano plot for schema
volc_limma_remote_control &lt;- ggplot(data=limma_remote_control, 
                                    aes(x= logFC, y= -log10(pval))) +
    geom_point(size = 3, color = &quot;black&quot;)+
  theme(axis.title = element_text(size =20),
        axis.text = element_blank(),
        axis.ticks = element_blank()) +
  geom_vline(xintercept = 0, linetype = 2)

volc_limma_remote_control</code></pre>
<pre><code>Warning: Removed 114 rows containing missing values (`geom_point()`).</code></pre>
<p><img src="figure/figures.Figure4.DVP.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>save_plot(filename = &quot;./plots/Figure_4.volcano_schema.pdf&quot;,
          plot = volc_limma_remote_control,
          base_asp = 1.4,
          base_height = 3)</code></pre>
<pre><code>Warning: Removed 114 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div id="subfigure-d---pathway-enrichment-for-mi_iz-vs-mi_remote"
class="section level1">
<h1>Subfigure D - Pathway enrichment for MI_IZ vs MI_remote</h1>
<pre class="r"><code>sig_pathways_mi &lt;- subset(mi_pathways,p_value &lt;= 0.05) %&gt;%
  arrange(desc(score)) %&gt;%
  dplyr::select(-statistic,-condition) %&gt;%
  subset(score &gt; 3 | score &lt; -3)

sig_pathways_mi$source &lt;- gsub(&quot;HALLMARK_&quot;,&quot;&quot;,sig_pathways_mi$source)

sig_pathways_mi$source &lt;- gsub(&quot;_&quot;,&quot; &quot;,sig_pathways_mi$source)
path_plot &lt;- ggplot(sig_pathways_mi, aes(x = reorder(source, score), y = score)) + 
    geom_bar(aes(fill = score),color = &quot;black&quot;, stat = &quot;identity&quot;) +
    scale_fill_gradient2(low = &quot;darkorange&quot;, high = &quot;purple&quot;,
        mid = &quot;white&quot;, midpoint = 0) +
  # scale_fill_viridis(option = &quot;F&quot;, direction = 1) +
    theme(axis.title = element_text(face = &quot;bold&quot;, size = 20),
          axis.text.x = element_text(hjust = 1, size =20, face= &quot;bold&quot;),
          axis.text.y = element_text(size =16),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          legend.text = element_text(size =20),
          legend.title = element_text(size =20)) +
    xlab(&quot;Pathways&quot;) +
  coord_flip()

path_plot</code></pre>
<p><img src="figure/figures.Figure4.DVP.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>save_plot(filename = &quot;./plots/Figure_5.pathway_plot.pdf&quot;,
          plot = path_plot,
          base_asp = 2,
          base_height = 4)</code></pre>
</div>
<div id="subfigure-e---vwf-specificity-for-endocardial-cells"
class="section level1">
<h1>Subfigure E - Vwf specificity for Endocardial cells</h1>
<pre class="r"><code>snrna_prot &lt;- fread(&quot;./output/proteomics/proteomics.snRNAseq_comp.tsv&quot;)
snrna_prot &lt;- snrna_prot %&gt;%
  mutate(&quot;label_gene&quot; = if_else(gene %in% c(&quot;Cdh11&quot;,&quot;Thbd&quot;,&quot;Vcam1&quot;),gene,
                                if_else(gene == &quot;Vwf&quot;,&quot;vWF&quot;,&quot;&quot;))) %&gt;%
  subset(pct.1 &gt; 0.05)

endo_proteomic_corr &lt;- ggplot(snrna_prot,aes(avg_log2FC,logFC,
                              label = label_gene)) +
  geom_hline(yintercept = 0, linetype = 2) +
  geom_point(data =subset(snrna_prot,gene != &quot;Vwf&quot;), size =3, fill = &quot;darkgrey&quot;, pch = 21) +
  geom_point(data = subset(snrna_prot,gene %in% c(&quot;Vwf&quot;,&quot;Vcam1&quot;)),size = 4, fill = &quot;purple&quot;, pch = 21) +
  geom_point(data = subset(snrna_prot,gene %in% c(&quot;Thbd&quot;)),size = 4, fill = &quot;darkorange&quot;, pch = 21) +
    geom_point(data = subset(snrna_prot,gene %in% c(&quot;Cdh11&quot;)),size = 4, fill = &quot;grey20&quot;, pch = 21) +
  geom_label_repel(size = 5.5, max.overlaps = 20,force = 3) +
  labs(x = &quot;Specificity endocard. cells (snRNA-seq)&quot;,
       y = &quot;Log2 fold-change (Proteomics)&quot;) 

endo_proteomic_corr</code></pre>
<pre><code>Warning: Removed 1598 rows containing missing values (`geom_point()`).</code></pre>
<pre><code>Warning: Removed 1598 rows containing missing values (`geom_label_repel()`).</code></pre>
<p><img src="figure/figures.Figure4.DVP.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>save_plot(filename = &quot;./plots/Figure_5.vwf_specificity_plot.pdf&quot;,
          plot = endo_proteomic_corr,
          base_asp = 1.75,
          base_height = 3.5)</code></pre>
<pre><code>Warning: Removed 1598 rows containing missing values (`geom_point()`).
Removed 1598 rows containing missing values (`geom_label_repel()`).</code></pre>
</div>
<div id="subfigure-f---vwf-is-upregulated-in-mi_iz"
class="section level1">
<h1>Subfigure F - Vwf is upregulated in MI_IZ</h1>
<pre class="r"><code>source(&quot;./code/functions.R&quot;)
yaxis_limits &lt;- c(11,17)

vsn_matrix &lt;- fread(&quot;./output/proteomics/proteomics.vsn_norm_proteins.tsv&quot;)
colnames(vsn_matrix)[1:11] &lt;- paste(&quot;s&quot;,1:11,colnames(vsn_matrix)[1:11],sep=&quot;_&quot;)
protein_sub &lt;- vsn_matrix  %&gt;%
    dplyr::select(1:11,gene) %&gt;%
    pivot_longer(1:11,names_to = &quot;sample&quot;, values_to = &quot;exp&quot;) %&gt;%
    mutate(&quot;group&quot; = if_else(grepl(&quot;control&quot;,sample),&quot;control&quot;,
                                   if_else(grepl(&quot;MI_IZ&quot;,sample),
                                           &quot;MI_IZ&quot;,&quot;MI_remote&quot;))
    )

protein_sub$group &lt;- gsub(&quot;control&quot;,&quot;Control&quot;,protein_sub$group)

protein_sub$group &lt;- factor(protein_sub$group,
                              levels = c(&quot;Control&quot;,&quot;MI_remote&quot;,&quot;MI_IZ&quot;))


## Barplot with points as alternative.
# goi &lt;- &quot;Vwf&quot;
# vwf_plot_bar &lt;- plot_proteomics_boxplot(norm_table = protein_sub,
#                                     protein = goi,
#                                     style = &quot;bar&quot;) +
#   geom_signif(comparisons = list(c(&quot;MI_IZ&quot;,&quot;MI_remote&quot;)),
#               tip_length = 0, annotation = &quot;0.0057&quot;, y_position = 15.5) +
#   geom_signif(comparisons = list(c(&quot;MI_IZ&quot;,&quot;Control&quot;)),
#               tip_length = 0, annotation = &quot;0.0022&quot;, y_position = 16.5) +
#    expand_limits(y = c(13, 17)) +
#   theme(axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) +
#   labs(x = &quot;&quot;)


## Median plot with points
goi &lt;- &quot;Cdh11&quot;
npr3_plot &lt;- plot_proteomics_boxplot(norm_table = protein_sub,
                                    protein = goi,
                                    style = &quot;mean&quot;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) +
  labs(x = &quot;&quot;) +
  ylim(yaxis_limits) +
  labs(y = &quot;&quot;) +
  scale_x_discrete(labels=c(&quot;Control&quot; = &quot;Control&quot;, 
                            &quot;MI_remote&quot; = &quot;MI remote&quot;,
                             &quot;MI_IZ&quot; = &quot;MI IZ&quot;))</code></pre>
<pre><code>Warning: The `fun.y` argument of `stat_summary()` is deprecated as of ggplot2 3.3.0.
ℹ Please use the `fun` argument instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<pre class="r"><code>## Mean plot with points
goi &lt;- &quot;Thbd&quot;
thbd_plot &lt;- plot_proteomics_boxplot(norm_table = protein_sub,
                                    protein = goi,
                                    style = &quot;mean&quot;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) +
  geom_signif(comparisons = list(c(&quot;MI_IZ&quot;,&quot;MI_remote&quot;)),
              tip_length = 0, annotation = &quot;0.001&quot;, y_position = 16.5) +
  geom_signif(comparisons = list(c(&quot;MI_IZ&quot;,&quot;Control&quot;)),
              tip_length = 0, annotation = &quot;0.023&quot;, y_position = 15.5) +
  labs(x = &quot;&quot;) +
  ylim(yaxis_limits) + 
  labs(y = &quot;&quot;) +
  scale_x_discrete(labels=c(&quot;Control&quot; = &quot;Control&quot;, 
                            &quot;MI_remote&quot; = &quot;MI remote&quot;,
                             &quot;MI_IZ&quot; = &quot;MI IZ&quot;))

## Median plot with points
goi &lt;- &quot;Vcam1&quot;
vcam1_plot &lt;- plot_proteomics_boxplot(norm_table = protein_sub,
                                    protein = goi,
                                    style = &quot;mean&quot;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) +
  geom_signif(comparisons = list(c(&quot;MI_IZ&quot;,&quot;MI_remote&quot;)),
              tip_length = 0, annotation = &quot;6.7e-4&quot;, y_position = 16.5) +
  geom_signif(comparisons = list(c(&quot;MI_IZ&quot;,&quot;Control&quot;)),
              tip_length = 0, annotation = &quot;0.0078&quot;, y_position = 15.5) +
  labs(x = &quot;&quot;) +
  ylim(yaxis_limits) + 
  labs(y = &quot;&quot;) +
  scale_x_discrete(labels=c(&quot;Control&quot; = &quot;Control&quot;, 
                            &quot;MI_remote&quot; = &quot;MI remote&quot;,
                             &quot;MI_IZ&quot; = &quot;MI IZ&quot;))


## Median plot with points
goi &lt;- &quot;Vwf&quot;
vwf_plot &lt;- plot_proteomics_boxplot(norm_table = protein_sub,
                                    protein = goi,
                                    style = &quot;mean&quot;) +
  geom_signif(comparisons = list(c(&quot;MI_IZ&quot;,&quot;MI_remote&quot;)),
              tip_length = 0, annotation = &quot;0.0057&quot;, y_position = 16.5) +
  geom_signif(comparisons = list(c(&quot;MI_IZ&quot;,&quot;Control&quot;)),
              tip_length = 0, annotation = &quot;0.0022&quot;, y_position = 15.5) +
  theme(axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) +
  labs(x = &quot;&quot;) +
  ylim(yaxis_limits) +
  scale_x_discrete(labels=c(&quot;Control&quot; = &quot;Control&quot;, 
                            &quot;MI_remote&quot; = &quot;MI remote&quot;,
                             &quot;MI_IZ&quot; = &quot;MI IZ&quot;))

# save_plot(filename = &quot;./figures/Figure_5.vwf_expression_plot.pdf&quot;,
#           plot = vwf_plot,
#           base_asp = 0.5,
#           base_height = 4)


joined_plot &lt;- npr3_plot + thbd_plot +  vcam1_plot + vwf_plot + plot_layout(nrow = 1,axis_titles = &quot;collect&quot;) &amp; labs(y = &quot;Normalized protein level&quot;)

save_plot(filename = &quot;./plots/Figure_5.expression_plot.pdf&quot;,
          plot = joined_plot,
          base_asp = 2.5,
          base_height = 4)</code></pre>
<pre><code>Warning: The dot-dot notation (`..y..`) was deprecated in ggplot2 3.4.0.
ℹ Please use `after_stat(y)` instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
</div>
<div id="combine-plots-for-figure" class="section level1">
<h1>Combine plots for figure</h1>
<pre class="r"><code># # Combine plots
# comb_plot &lt;- (pca_plot + volc_limma_IZ_remote + path_plot) / (endo_proteomic_corr + wrap_plots(npr3_plot,vwf_plot))
# 
# save_plot(filename = &quot;./figures/Figure_4.proteomics_combined.pdf&quot;,
#           plot = comb_plot,
#           base_asp = 2.5,
#           base_height = 12)</code></pre>
</div>
<div id="subfigure-h---human-cite-seq-umap" class="section level1">
<h1>Subfigure H - human CITE-seq umap</h1>
<pre class="r"><code>library(Seurat)</code></pre>
<pre><code>Loading required package: SeuratObject</code></pre>
<pre><code>Loading required package: sp</code></pre>
<pre><code>&#39;SeuratObject&#39; was built with package &#39;Matrix&#39; 1.6.3 but the current
version is 1.6.5; it is recomended that you reinstall &#39;SeuratObject&#39; as
the ABI for &#39;Matrix&#39; may have changed</code></pre>
<pre><code>
Attaching package: &#39;SeuratObject&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    intersect</code></pre>
<pre class="r"><code>library(SCpubr)</code></pre>
<pre><code>

── SCpubr 2.0.0.9000 ───────────────────────────────────────────────────────────

ℹ Have a look at extensive tutorials in SCpubr&#39;s book.

✔ If you use SCpubr in your research, please cite it accordingly.

★ If the package is useful to you, consider leaving a Star in the GitHub repository.

! Keep track of the package updates on Twitter (@Enblacar) or in the Official NEWS website.

♥ Happy plotting!



── Package version ──

CRAN:           2.0.2
Installed: 2.0.0.9000

⚠ There is a new version available onCRAN!



── Required packages ──

✔ AnnotationDbi       1.64.1 | 1.58.0         ✔ assertthat           0.2.1 | 0.2.1          ✖ AUCell                                 
✔ circlize            0.4.15 | 0.4.16         ✔ cluster              2.1.6 | 2.1.6          ✖ clusterProfiler                        
✔ colorspace           2.1.0 | 2.1-0          ✔ decoupleR            2.8.0 | 2.2.2          ✔ dplyr                1.1.4 | 1.1.4     
✖ enrichplot                                  ✔ forcats              1.0.0 | 1.0.0          ✖ ggalluvial                             
✔ ggbeeswarm           0.7.2 | 0.7.2          ✖ ggdist                                      ✖ ggExtra                                
✖ ggnewscale                                  ✔ ggplot2              3.4.4 | 3.5.0          ✔ ggplotify            0.1.2 | 0.1.2     
✔ ggrastr              1.0.2 | 1.0.2          ✔ ggrepel              0.9.5 | 0.9.5          ✔ ggridges             0.5.5 | 0.5.6     
✔ ggsignif             0.6.4 | 0.6.4          ✔ labeling             0.4.3 | 0.4.3          ✖ liana                                  
✔ magrittr             2.0.3 | 2.0.3          ✔ MASS            7.3.60.0.1 | 7.3-60.0.1     ✔ Matrix               1.6.5 | 1.6-5     
✔ Nebulosa            1.12.0 | 1.6.0          ✔ patchwork            1.2.0 | 1.2.0          ✔ pbapply              1.7.2 | 1.7-2     
✔ plyr                 1.8.9 | 1.8.9          ✔ RColorBrewer         1.1.3 | 1.1-3          ✔ rlang                1.1.3 | 1.1.3     
✔ scales               1.3.0 | 1.3.0          ✔ scattermore            1.2 | 1.2            ✔ Seurat               5.0.1 | 5.0.3     
✔ SeuratObject         5.0.1 | 5.0.1          ✔ stringr              1.5.1 | 1.5.1          ✔ svglite              2.1.3 | 2.1.3     
✔ tibble               3.2.1 | 3.2.1          ✔ tidyr                1.3.0 | 1.3.1          ✖ UCell                                  
✔ viridis              0.6.4 | 0.6.5          ✔ withr                2.5.2 | 3.0.0     

ℹ Installed packages are denoted by a tick (✔) and missing packages by a cross (✖).
ℹ Installed packages that still require an update to correctly run SCpubr have an exclamation mark (!).
ℹ Packages version are displayed as: Installed | Available.



── Available functions ──

✔ do_AffinityAnalysisPlot | DEV     ✖ do_AlluvialPlot                   ✔ do_BarPlot                   
✔ do_BeeSwarmPlot                   ✔ do_BoxPlot                        ✖ do_CellularStatesPlot        
✔ do_ChordDiagramPlot               ✔ do_ColorPalette                   ✖ do_CopyNumberVariantPlot     
✔ do_CorrelationPlot                ✔ do_DiffusionMapPlot | DEV         ✔ do_DimPlot                   
✔ do_DotPlot                        ✖ do_EnrichmentHeatmap              ✔ do_ExpressionHeatmap         
✔ do_FeaturePlot                    ✖ do_FunctionalAnnotationPlot       ✖ do_GeyserPlot                
✖ do_GroupedGOTermPlot              ✔ do_GroupwiseDEPlot                ✖ do_LigandReceptorPlot | DEV  
✔ do_LoadingsPlot                   ✔ do_MetadataPlot | DEV             ✔ do_NebulosaPlot              
✔ do_PathwayActivityPlot            ✔ do_RidgePlot                      ✖ do_SCEnrichmentHeatmap | DEV 
✔ do_SCExpressionHeatmap | DEV      ✔ do_TermEnrichmentPlot             ✔ do_TFActivityPlot            
✔ do_ViolinPlot                     ✔ do_VolcanoPlot                    ✔ save_Plot | DEV              

ℹ Functions tied to development builds of SCpubr are marked by the (| DEV) tag.
ℹ You can install development builds of SCpubr by following the instructions in the Releases page.
ℹ Check the package requirements function-wise with: SCpubr::check_dependencies()



── Tips! ──

ℹ To adjust package messages to dark mode themes, use: options(&quot;SCpubr.darkmode&quot; = TRUE)
ℹ To remove the white and black end from continuous palettes, use: options(&quot;SCpubr.ColorPaletteEnds&quot; = FALSE)

✖ To suppress this startup message, use: suppressPackageStartupMessages(library(SCpubr))
✖ Alternatively, you can also set the following option: options(&quot;SCpubr.verbose&quot; = FALSE)
  And then load the package normally (and faster) as: library(SCpubr)

────────────────────────────────────────────────────────────────────────────────</code></pre>
<pre class="r"><code>human_citeseq &lt;- readRDS(&quot;../public_data/Amrute_et_al/final_global_annotated.rds&quot;)</code></pre>
<pre class="r"><code>DefaultAssay(human_citeseq) &lt;- &quot;SCT&quot;

#Choose endocardial cluster
Idents(human_citeseq) &lt;- &quot;annotation.0.1&quot;
human_endocardium &lt;- subset (human_citeseq, idents = &quot;Endocardium&quot;)

#Choose Donor and AMI only
Idents(human_endocardium) &lt;- &quot;HF.etiology&quot;
human_endocardium &lt;- subset (human_endocardium, idents = c(&quot;Donor&quot;, &quot;AMI&quot;))

#plot VWF expression
plot3 &lt;- VlnPlot (human_endocardium, feature = c(&quot;VWF&quot;), cols = c(&quot;#008000&quot;, &quot;#CD1076&quot;))

#plot Umap embedding using SCpubr package
named_colors &lt;- c(&quot;Fibroblast&quot; = &quot;#1f77b4&quot;,
                    &quot;B Cells&quot; = &quot;#d62728&quot;,
                    &quot;Plasma Cells&quot; = &quot;#ff7f0e&quot;,
                    &quot;Endocardium&quot; = &quot;#17becf&quot;,
                    &quot;Endothelium&quot; = &quot;#8c564b&quot;,
                    &quot;Lymphatics&quot; = &quot;lightgrey&quot;,
                    &quot;T_NK Cells&quot; = &quot;#bcbd22&quot;,
                    &quot;Myeloid&quot; = &quot;#2ca02c&quot;,
                    &quot;Glia&quot; = &quot;#9467bd&quot;,
                    &quot;SMC_Pericyte&quot; = &quot;#e377c2&quot;, 
                    &quot;Mast Cells&quot; = &quot;darkred&quot;)

human_cite_umap &lt;- SCpubr::do_DimPlot(sample = human_citeseq, 
                            label = FALSE, label.box= TRUE, 
                            group.by = &quot;annotation.0.1&quot;, 
                            repel = TRUE, 
                            legend.position = &quot;right&quot;, plot_cell_borders = TRUE, 
                            plot_density_contour = FALSE,
                            plot.axes = FALSE, raster.dpi = 300, 
                            shuffle = FALSE, 
                            pt.size = 0.4, reduction = &quot;rna.umap&quot;,
                            legend.icon.size = 5, 
                            legend.byrow = TRUE, colors.use = named_colors) +
  theme(legend.position = &quot;none&quot;)

save_plot(human_cite_umap,
          file = &quot;./plots/Figure4.human_citeseq_umap.png&quot;,
          base_height = 3,
          base_asp = 1.3)</code></pre>
</div>
<div id="subfigure-i---violin-plot-for-vwf" class="section level1">
<h1>Subfigure I - Violin plot for vWF</h1>
<pre class="r"><code>sub_human_cite &lt;- subset(human_endocardium,HF.etiology %in% c(&quot;Donor&quot;,&quot;AMI&quot;))
sub_human_cite$disease_group &lt;- sub_human_cite$HF.etiology

## Quick DE analysis between Donor and AMI
sub_human_cite_pb &lt;- AggregateExpression(sub_human_cite,
                                   return.seurat = T,
                                   group.by = c(&quot;sample&quot;,&quot;disease_group&quot;))</code></pre>
<pre><code>Centering and scaling data matrix</code></pre>
<pre class="r"><code>Idents(sub_human_cite_pb) &lt;- &quot;disease_group&quot;

sub_human_cite_pb_de &lt;- FindMarkers(object = sub_human_cite_pb,
                         ident.1 = &quot;Donor&quot;,
                         ident.2 = &quot;AMI&quot;,
                         test.use = &quot;DESeq2&quot;,
                         min.pct = 0.1)</code></pre>
<pre><code>converting counts to integer mode</code></pre>
<pre><code>gene-wise dispersion estimates</code></pre>
<pre><code>mean-dispersion relationship</code></pre>
<pre><code>final dispersion estimates</code></pre>
<pre class="r"><code>sub_human_cite_pb_de$gene &lt;- rownames(sub_human_cite_pb_de)
sub_human_cite_pb_vwf &lt;- subset(sub_human_cite_pb_de,gene == &quot;VWF&quot;)
pvalue &lt;- sub_human_cite_pb_vwf$p_val
pvalue</code></pre>
<pre><code>[1] 1.405446e-05</code></pre>
<pre class="r"><code>Idents(sub_human_cite) &lt;- sub_human_cite$disease_group
vwf_vlnpot &lt;- SCpubr::do_ViolinPlot(sample = sub_human_cite, 
                      features = &quot;VWF&quot;,
                      group.by = &quot;disease_group&quot;,
                      line_width = 1,
                      legend.position = &quot;none&quot;,
                      legend.title = &quot;&quot;,
                      font.size = 25,
                      ylab = &quot;Expression level&quot;,
                      xlab = &quot;&quot;,
                      colors.use = c(&quot;Donor&quot; = &quot;#008000&quot;, 
                                     &quot;AMI&quot; = &quot;#CD1076&quot;,
                                     &quot;ICM&quot; = &quot;white&quot;,
                                     &quot;NICM&quot; = &quot;white&quot;))

 vwf_vlnpot &lt;- vwf_vlnpot + theme(plot.margin = margin(t=10, r=10, b=-25, l=10, unit=&quot;pt&quot;))

save_plot(vwf_vlnpot,
          file = &quot;./plots/Figure4.human_citeseq_vlnplot.pdf&quot;,
          base_height = 4)</code></pre>
</div>
<div id="save-differential-protein-expression-results-for-table-3"
class="section level1">
<h1>Save differential protein expression results for Table 3</h1>
<pre class="r"><code>table3 &lt;- limma_res %&gt;%
  select(-c(label_protein,&quot;P.Value&quot;))

colnames(table3) &lt;- gsub(&quot;\\.&quot;,&quot;_&quot;,colnames(table3))
colnames(table3) &lt;- gsub(&quot;adj_P_Val&quot;,&quot;ajusted_pval&quot;,colnames(table3))

table3 &lt;- table3 %&gt;%
  select(analysis,logFC,AveExpr,t,pval,ajusted_pval,B,gene,protein_ids) %&gt;%
  arrange(desc(logFC)) %&gt;%
  drop_na()

write.table(table3,
            file = &quot;./output/proteomics/Table3.tsv&quot;,
            sep = &quot;\t&quot;,
            quote = F,
            row.names = F,
            col.names = TRUE)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.1.2

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] SCpubr_2.0.0.9000  Seurat_5.0.1       SeuratObject_5.0.1 sp_2.1-2          
 [5] RColorBrewer_1.1-3 ggsci_3.0.0        cowplot_1.1.2      ggforce_0.4.1     
 [9] patchwork_1.2.0    ggsignif_0.6.4     ggbeeswarm_0.7.2   ggrepel_0.9.5     
[13] here_1.0.1         data.table_1.14.10 lubridate_1.9.3    forcats_1.0.0     
[17] stringr_1.5.1      dplyr_1.1.4        purrr_1.0.2        readr_2.1.5       
[21] tidyr_1.3.0        tibble_3.2.1       ggplot2_3.4.4      tidyverse_2.0.0   
[25] workflowr_1.7.1   

loaded via a namespace (and not attached):
  [1] fs_1.6.3                    matrixStats_1.2.0          
  [3] spatstat.sparse_3.0-3       bitops_1.0-7               
  [5] httr_1.4.7                  tools_4.3.1                
  [7] sctransform_0.4.1           utf8_1.2.4                 
  [9] R6_2.5.1                    lazyeval_0.2.2             
 [11] uwot_0.1.16                 withr_2.5.2                
 [13] gridExtra_2.3               progressr_0.14.0           
 [15] cli_3.6.2                   Biobase_2.62.0             
 [17] textshaping_0.3.7           spatstat.explore_3.2-5     
 [19] fastDummies_1.7.3           labeling_0.4.3             
 [21] sass_0.4.8                  mvtnorm_1.2-4              
 [23] spatstat.data_3.0-3         ggridges_0.5.5             
 [25] pbapply_1.7-2               systemfonts_1.0.5          
 [27] yulab.utils_0.1.3           svglite_2.1.3              
 [29] parallelly_1.36.0           rstudioapi_0.15.0          
 [31] RSQLite_2.3.4               generics_0.1.3             
 [33] gridGraphics_0.5-1          shape_1.4.6                
 [35] ica_1.0-3                   spatstat.random_3.2-2      
 [37] Matrix_1.6-5                fansi_1.0.6                
 [39] S4Vectors_0.40.2            abind_1.4-5                
 [41] lifecycle_1.0.4             whisker_0.4.1              
 [43] yaml_2.3.8                  SummarizedExperiment_1.32.0
 [45] SparseArray_1.2.3           Rtsne_0.17                 
 [47] grid_4.3.1                  blob_1.2.4                 
 [49] promises_1.2.1              crayon_1.5.2               
 [51] miniUI_0.1.1.1              lattice_0.22-5             
 [53] KEGGREST_1.42.0             pillar_1.9.0               
 [55] knitr_1.45                  GenomicRanges_1.54.1       
 [57] future.apply_1.11.1         codetools_0.2-19           
 [59] leiden_0.4.3.1              glue_1.7.0                 
 [61] getPass_0.2-4               vctrs_0.6.5                
 [63] png_0.1-8                   spam_2.10-0                
 [65] gtable_0.3.4                assertthat_0.2.1           
 [67] ks_1.14.2                   cachem_1.0.8               
 [69] xfun_0.41                   S4Arrays_1.2.0             
 [71] mime_0.12                   pracma_2.4.4               
 [73] survival_3.5-7              SingleCellExperiment_1.24.0
 [75] ellipsis_0.3.2              fitdistrplus_1.1-11        
 [77] ROCR_1.0-11                 nlme_3.1-164               
 [79] bit64_4.0.5                 RcppAnnoy_0.0.21           
 [81] GenomeInfoDb_1.38.5         rprojroot_2.0.4            
 [83] bslib_0.6.1                 irlba_2.3.5.1              
 [85] vipor_0.4.7                 KernSmooth_2.23-22         
 [87] colorspace_2.1-0            BiocGenerics_0.48.1        
 [89] DBI_1.2.0                   DESeq2_1.42.0              
 [91] ggrastr_1.0.2               tidyselect_1.2.0           
 [93] processx_3.8.3              bit_4.0.5                  
 [95] compiler_4.3.1              git2r_0.33.0               
 [97] DelayedArray_0.28.0         plotly_4.10.4              
 [99] scales_1.3.0                lmtest_0.9-40              
[101] callr_3.7.3                 digest_0.6.34              
[103] goftest_1.2-3               spatstat.utils_3.0-4       
[105] rmarkdown_2.25              XVector_0.42.0             
[107] decoupleR_2.8.0             htmltools_0.5.7            
[109] pkgconfig_2.0.3             MatrixGenerics_1.14.0      
[111] highr_0.10                  fastmap_1.1.1              
[113] rlang_1.1.3                 GlobalOptions_0.1.2        
[115] htmlwidgets_1.6.4           shiny_1.8.0                
[117] farver_2.1.1                jquerylib_0.1.4            
[119] zoo_1.8-12                  jsonlite_1.8.8             
[121] BiocParallel_1.36.0         mclust_6.0.1               
[123] RCurl_1.98-1.14             magrittr_2.0.3             
[125] GenomeInfoDbData_1.2.11     ggplotify_0.1.2            
[127] dotCall64_1.1-1             munsell_0.5.0              
[129] Rcpp_1.0.12                 viridis_0.6.4              
[131] reticulate_1.34.0           stringi_1.8.3              
[133] zlibbioc_1.48.0             MASS_7.3-60.0.1            
[135] plyr_1.8.9                  parallel_4.3.1             
[137] listenv_0.9.0               deldir_2.0-2               
[139] Biostrings_2.70.1           splines_4.3.1              
[141] tensor_1.5                  hms_1.1.3                  
[143] circlize_0.4.15             locfit_1.5-9.8             
[145] ps_1.7.6                    igraph_1.6.0               
[147] spatstat.geom_3.2-7         RcppHNSW_0.5.0             
[149] reshape2_1.4.4              stats4_4.3.1               
[151] evaluate_0.23               Nebulosa_1.12.0            
[153] renv_1.0.3                  BiocManager_1.30.22        
[155] tzdb_0.4.0                  tweenr_2.0.2               
[157] httpuv_1.6.14               RANN_2.6.1                 
[159] polyclip_1.10-6             future_1.33.1              
[161] scattermore_1.2             xtable_1.8-4               
[163] RSpectra_0.16-1             later_1.3.2                
[165] viridisLite_0.4.2           ragg_1.2.7                 
[167] memoise_2.0.1               beeswarm_0.4.0             
[169] AnnotationDbi_1.64.1        IRanges_2.36.0             
[171] cluster_2.1.6               timechange_0.2.0           
[173] globals_0.16.2             </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
