<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="FloWuenne" />

<meta name="date" content="2023-06-26" />

<title>figures.supplementary_figure_X.proteomics_qc</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mi_spatialomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="data_processing.html">Data processing</a>
</li>
<li>
  <a href="data_analysis.html">Data analysis</a>
</li>
<li>
  <a href="figures.html">Figures</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">figures.supplementary_figure_X.proteomics_qc</h1>
<h4 class="author">FloWuenne</h4>
<h4 class="date">2023-06-26</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-03-21
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>mi_spatialomics/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230612code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230612)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230612code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230612)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomFloWuennemispatialomicstreee6213a5ae03dcd38ce227600f6ad3c4c9a7c5fc0targetblanke6213a5a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/FloWuenne/mi_spatialomics/tree/e6213a5ae03dcd38ce227600f6ad3c4c9a7c5fc0" target="_blank">e6213a5</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomFloWuennemispatialomicstreee6213a5ae03dcd38ce227600f6ad3c4c9a7c5fc0targetblanke6213a5a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/FloWuenne/mi_spatialomics/tree/e6213a5ae03dcd38ce227600f6ad3c4c9a7c5fc0" target="_blank">e6213a5</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/deprecated/.DS_Store
    Ignored:    analysis/molecular_cartography_python/.DS_Store
    Ignored:    analysis/seqIF_python/.DS_Store
    Ignored:    analysis/seqIF_python/pixie/.DS_Store
    Ignored:    analysis/seqIF_python/pixie/cell_clustering/
    Ignored:    annotations/.DS_Store
    Ignored:    annotations/SeqIF/.DS_Store
    Ignored:    annotations/molkart/.DS_Store
    Ignored:    annotations/molkart/Figure1_regions/.DS_Store
    Ignored:    annotations/molkart/Supplementary_Figure4_regions/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/140623.calcagno_et_al.seurat_object.rds
    Ignored:    data/Calcagno2022_int_logNorm_annot.h5Seurat
    Ignored:    data/IC_03_IF_CCR2_CD68 cell numbers.xlsx
    Ignored:    data/Traditional_IF_absolute_cell_counts.csv
    Ignored:    data/Traditional_IF_relative_cell_counts.csv
    Ignored:    data/pixie.cell_table_size_normalized_cell_labels.csv
    Ignored:    data/results_cts_100.sqm
    Ignored:    data/seqIF_regions_annotations/
    Ignored:    data/seurat/
    Ignored:    output/.DS_Store
    Ignored:    output/mol_cart.harmony_object.h5Seurat
    Ignored:    output/molkart/
    Ignored:    output/proteomics/
    Ignored:    output/results_cts.lowres.125.sqm
    Ignored:    output/seqIF/
    Ignored:    pipeline_configs/.DS_Store
    Ignored:    plots/
    Ignored:    references/.DS_Store
    Ignored:    renv/.DS_Store
    Ignored:    renv/library/
    Ignored:    renv/staging/

Untracked files:
    Untracked:  analysis/deprecated/figures.supplementary_figureX.Rmd
    Untracked:  analysis/deprecated/figures.supplementary_figure_X.MistyR.Rmd

Unstaged changes:
    Deleted:    analysis/figures.supplementary_figureX.Rmd
    Deleted:    analysis/figures.supplementary_figure_X.MistyR.Rmd
    Deleted:    analysis/figures.supplementary_figure_X.proteomics_qc.Rmd
    Deleted:    figures/Figure_5.eps
    Deleted:    figures/Figure_5.pdf
    Deleted:    figures/Figure_5.png
    Deleted:    figures/Figure_5.svg
    Deleted:    figures/Supplementary_Figure_1_Molecular_Cartography_ROIs.png
    Deleted:    figures/Supplementary_figure_5.segmentation_metrics.poster.eps
    Modified:   figures/Supplementary_figure_X.proteomics.eps
    Modified:   figures/Supplementary_figure_X.proteomics.png
    Deleted:    results_cts.lowres.125.sqm

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/figures.supplementary_figure_9.proteomics_qc.Rmd</code>)
and HTML
(<code>docs/figures.supplementary_figure_9.proteomics_qc.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/e6213a5ae03dcd38ce227600f6ad3c4c9a7c5fc0/analysis/figures.supplementary_figure_9.proteomics_qc.Rmd" target="_blank">e6213a5</a>
</td>
<td>
FloWuenne
</td>
<td>
2024-03-21
</td>
<td>
Publish all updated files.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This supplementary figure contains QC plots for the proteomic data
analysis.</p>
<pre class="r"><code>## load data
metadata_counts &lt;- fread(&quot;./output/proteomics/proteomics.protein_missing_stats.tsv&quot;)</code></pre>
<div id="subfigure-a---total-proteins-identified-per-group"
class="section level1">
<h1>Subfigure A - Total proteins identified per group</h1>
<pre class="r"><code>## Summarize observations across group
metadata_counts_stat &lt;- data_summary(metadata_counts,
                                     varname = &quot;wo_cont&quot;, groupnames = c(&quot;group&quot;))</code></pre>
<pre><code>Loading required package: plyr</code></pre>
<pre><code>------------------------------------------------------------------------------</code></pre>
<pre><code>You have loaded plyr after dplyr - this is likely to cause problems.
If you need functions from both plyr and dplyr, please load plyr first, then dplyr:
library(plyr); library(dplyr)</code></pre>
<pre><code>------------------------------------------------------------------------------</code></pre>
<pre><code>
Attaching package: &#39;plyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:plotly&#39;:

    arrange, mutate, rename, summarise</code></pre>
<pre><code>The following object is masked from &#39;package:here&#39;:

    here</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    arrange, count, desc, failwith, id, mutate, rename, summarise,
    summarize</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    compact</code></pre>
<pre class="r"><code>avg_proteins &lt;- mean(metadata_counts$wo_cont)
avg_proteins &lt;- sd(metadata_counts$wo_cont)

metadata_counts_stat$group &lt;- factor(metadata_counts_stat$group,
                                     levels = c(&quot;MI_IZ&quot;,&quot;MI_remote&quot;,&quot;control&quot;))

## Barplot to show the number of proteins detected per group (excluding contaminants)
n_prot_plot &lt;- ggplot(metadata_counts_stat,aes(group,wo_cont)) +
  geom_bar(aes(fill = group),stat=&quot;identity&quot;, color=&quot;black&quot;, 
           position=position_dodge()) +
  geom_point(data = metadata_counts, aes(group,wo_cont),
             fill = &quot;white&quot;,pch = 21, color= &quot;black&quot;,size = 4) +
  geom_errorbar(aes(ymin=wo_cont, ymax=wo_cont+sd), width=.2,
                 position=position_dodge(.9)) + 
  coord_flip() +
  scale_fill_manual(values = proteome_palette,
                    labels = c(&quot;Control&quot;,&quot;MI_remote&quot;,&quot;MI_IZ&quot;)) +
  labs(x = &quot;&quot;,
       y = &quot;Proteins detected&quot;) +
  theme(legend.position = &quot;none&quot;)

n_prot_plot</code></pre>
<p><img src="figure/figures.supplementary_figure_9.proteomics_qc.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="subfigure-b---missing-proteins-vs-number-of-nuclei-collected"
class="section level1">
<h1>Subfigure B - % missing proteins vs number of nuclei collected</h1>
<pre class="r"><code>metadata_counts$group &lt;- factor(metadata_counts$group,
                                levels = c(&quot;control&quot;,&quot;MI_remote&quot;,&quot;MI_IZ&quot;))

missingness_plot &lt;- ggplot(metadata_counts,aes(missingness,n_nuclei, label = sample)) +
  geom_point(pch = 21, size = 5,aes(fill = group)) +
  scale_fill_manual(values = proteome_palette,
                    labels = c(&quot;Control&quot;,&quot;MI_remote&quot;,&quot;MI_IZ&quot;)) +
  geom_label_repel() +
  labs(x = &quot;% missing values&quot;,
       y = &quot;# nuclei in tissue collected&quot;,
       fill = &quot;Group&quot;) +
  theme(legend.position = &quot;top&quot;)
missingness_plot</code></pre>
<p><img src="figure/figures.supplementary_figure_9.proteomics_qc.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="subfigure-c---variance-explained-by-pcs"
class="section level1">
<h1>Subfigure C - % variance explained by PCs</h1>
<pre class="r"><code>pca_res &lt;- readRDS(&quot;./output/proteomics/proteomics.pca_res.rds&quot;)

sdev &lt;- pca_res$sdev[1:5]
var_explained &lt;- data.frame(PC = paste0(&quot;PC&quot;,1:5),
                            var_expl = (sdev)^2/sum((sdev)^2)
                            )

pca_var &lt;- ggplot(var_explained,aes(PC,var_expl)) +
  geom_bar(stat = &quot;identity&quot;, color = &quot;black&quot;, fill = &quot;grey30&quot;) +
  geom_text(aes(label = paste(100*round(var_expl,2),sep = &quot;&quot;)), vjust=1.4, color = &quot;white&quot;, size =6) +
  labs(x = &quot;Principal components&quot;,
       y = &quot;% Variance explained&quot;)

pca_var</code></pre>
<p><img src="figure/figures.supplementary_figure_9.proteomics_qc.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="subfigure-d---upset-plot-for-shared-proteins-between-conditions"
class="section level1">
<h1>Subfigure D - Upset plot for shared proteins between conditions</h1>
<pre class="r"><code>library(ggplot2)
library(ComplexUpset)

## Load limma results and subset for significantly differential proteins between conditions
limma_res &lt;- fread(&quot;./output/proteomics/proteomics.limma.full_statistics.tsv&quot;)
limma_res_sig &lt;- subset(limma_res,adj.P.Val &lt;= 0.05)
limma_res_sig &lt;- limma_res_sig %&gt;%
  select(gene,analysis) %&gt;%
  mutate(value = 1) %&gt;%
  # Reshape the data frame so that each gene has 1 row entry for each comparison
  pivot_wider(names_from = analysis, 
              values_from = value, 
              values_fill = list(value = 0)) %&gt;%
  # Sum rows by gene
  group_by(gene) %&gt;%
  summarise_all(sum) %&gt;%
  ungroup()
              
analysis_groups &lt;- c(&quot;MI_IZ_vs_control&quot;,&quot;MI_remote_vs_control&quot;,&quot;MI_IZ_vs_MI_remote&quot;)
upset_plot &lt;- upset(limma_res_sig, analysis_groups , name=&#39;Comparison&#39;,
                    width_ratio=0.1,
                    set_sizes=(
                      upset_set_size()
        + theme(axis.text.x=element_text(angle=90))
    ))
upset_plot</code></pre>
<p><img src="figure/figures.supplementary_figure_9.proteomics_qc.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="assemble-final-figure" class="section level1">
<h1>Assemble final figure</h1>
<pre class="r"><code>supp_figure_X &lt;- n_prot_plot + missingness_plot + pca_var + wrap_elements(upset_plot)

supp_figure_X &lt;- supp_figure_X + 
  plot_layout(nrow = 2,ncol =2 , widths = c(1,2)) +
  plot_annotation(tag_levels = &#39;a&#39;) &amp; 
  theme(plot.tag = element_text(size = 25)) &amp;
  theme(plot.background = element_rect(fill = &quot;white&quot;))

supp_figure_X</code></pre>
<p><img src="figure/figures.supplementary_figure_9.proteomics_qc.Rmd/unnamed-chunk-6-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>save_plot(filename = &quot;./figures/Supplementary_figure_X.proteomics.png&quot;,
          plot = supp_figure_X,
          base_height = 8)


save_plot(filename = &quot;./figures/Supplementary_figure_X.proteomics.eps&quot;,
          plot = supp_figure_X,
          base_height = 8)</code></pre>
<!-- # Volcano plot of MI_IZ vs control -->
<!-- ```{r} -->
<!-- iz_control <- subset(limma_res, analysis == "MI_IZ_vs_control") -->
<!-- iz_control <- iz_control %>% -->
<!--   arrange(desc(logFC)) -->
<!-- prot_2label <- "" -->
<!-- prot_2label <- subset(iz_control, adj.P.Val < 0.05)[1:10,]$gene -->
<!-- prot_2label <- c(prot_2label,tail(subset(iz_control, adj.P.Val < 0.05),1)$gene) -->
<!-- iz_control <- iz_control %>% -->
<!--   mutate("label_protein" = if_else(gene %in% prot_2label, gene,"")) -->
<!-- ## Volcano plot of IZ_remote vs control -->
<!-- iz_control_volc <- plot_pretty_volcano(iz_control,  -->
<!--                                       pt_size = 2, -->
<!--                                       plot_title = "MI_remote vs control", -->
<!--                                       sig_col = "adj.P.Val", -->
<!--                                       sig_thresh = 0.05, -->
<!--                                       col_pos_logFC = proteome_palette[['control']], -->
<!--                                       col_neg_logFC = proteome_palette[['MI_IZ']]) + -->
<!--   geom_label_repel(max.overlaps = 10) -->
<!-- iz_control_volc -->
<!-- ``` -->
<!-- # Volcano plot of MI_remote vs control -->
<!-- ```{r} -->
<!-- remote_control <- subset(limma_res, analysis == "MI_remote_vs_control") -->
<!-- remote_control <- remote_control %>% -->
<!--   arrange(desc(logFC)) -->
<!-- prot_2label <- subset(remote_control, adj.P.Val < 0.05)[1:13,]$gene -->
<!-- prot_2label <- c(prot_2label,tail(subset(remote_control, adj.P.Val < 0.05),1)$gene) -->
<!-- remote_control <- remote_control %>% -->
<!--   mutate("label_protein" = if_else(gene %in% prot_2label, gene,"")) -->
<!-- ## Volcano plot of IZ_remote vs control -->
<!-- remote_control_volc <- plot_pretty_volcano(remote_control,  -->
<!--                                       pt_size = 2, -->
<!--                                       plot_title = "MI_remote vs control", -->
<!--                                       sig_col = "adj.P.Val", -->
<!--                                       sig_thresh = 0.05, -->
<!--                                       col_pos_logFC = proteome_palette[['MI_remote']], -->
<!--                                       col_neg_logFC = proteome_palette[['control']]) + -->
<!--   geom_label_repel() -->
<!-- remote_control_volc -->
<!-- ``` -->
<!-- # Plot candidate protein expressions -->
<!-- ```{r} -->
<!-- vsn_matrix <- fread("./output/proteomics/proteomics.vsn_norm_proteins.tsv") -->
<!-- protein_sub <- vsn_matrix  %>% -->
<!--     dplyr::select(1:11,gene) %>% -->
<!--     pivot_longer(1:11,names_to = "sample", values_to = "exp") %>% -->
<!--     mutate("group" = if_else(grepl("control",sample),"control", -->
<!--                                    if_else(grepl("MI_IZ",sample), -->
<!--                                            "MI_IZ","MI_remote")) -->
<!--     ) -->
<!-- protein_sub$group <- factor(protein_sub$group, -->
<!--                               levels = c("control","MI_remote","MI_IZ")) -->
<!-- goi <- "Vwf" -->
<!-- plot_proteomics_boxplot(norm_table = protein_sub, -->
<!--                         protein = goi) + -->
<!--   geom_signif(comparisons = list(c("MI_IZ","MI_remote")), -->
<!--               tip_length = 0, annotation = subset(limma_res, -->
<!--                                                   analysis == "MI_IZ_vs_MI_remote" & -->
<!--                                                     gene == goi)$adj.P.Val) -->
<!-- ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.1.2

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] ComplexUpset_1.3.3  plyr_1.8.9          RColorBrewer_1.1-3 
 [4] ggsci_3.0.0         cowplot_1.1.2       ggbeeswarm_0.7.2   
 [7] ggsignif_0.6.4      heatmaply_1.5.0     plotly_4.10.4      
[10] viridis_0.6.4       viridisLite_0.4.2   ggVennDiagram_1.5.0
[13] patchwork_1.2.0     ggrepel_0.9.5       here_1.0.1         
[16] data.table_1.14.10  lubridate_1.9.3     forcats_1.0.0      
[19] stringr_1.5.1       dplyr_1.1.4         purrr_1.0.2        
[22] readr_2.1.5         tidyr_1.3.0         tibble_3.2.1       
[25] ggplot2_3.4.4       tidyverse_2.0.0     workflowr_1.7.1    

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0    farver_2.1.1        vipor_0.4.7        
 [4] fastmap_1.1.1       TSP_1.2-4           lazyeval_0.2.2     
 [7] promises_1.2.1      digest_0.6.34       timechange_0.2.0   
[10] lifecycle_1.0.4     processx_3.8.3      magrittr_2.0.3     
[13] compiler_4.3.1      rlang_1.1.3         sass_0.4.8         
[16] tools_4.3.1         utf8_1.2.4          yaml_2.3.8         
[19] knitr_1.45          labeling_0.4.3      htmlwidgets_1.6.4  
[22] registry_0.5-1      ca_0.71.1           withr_2.5.2        
[25] grid_4.3.1          fansi_1.0.6         git2r_0.33.0       
[28] colorspace_2.1-0    scales_1.3.0        iterators_1.0.14   
[31] cli_3.6.2           rmarkdown_2.25      ragg_1.2.7         
[34] generics_0.1.3      rstudioapi_0.15.0   httr_1.4.7         
[37] tzdb_0.4.0          cachem_1.0.8        assertthat_0.2.1   
[40] BiocManager_1.30.22 vctrs_0.6.5         webshot_0.5.5      
[43] jsonlite_1.8.8      seriation_1.5.4     callr_3.7.3        
[46] hms_1.1.3           beeswarm_0.4.0      systemfonts_1.0.5  
[49] dendextend_1.17.1   foreach_1.5.2       jquerylib_0.1.4    
[52] glue_1.7.0          codetools_0.2-19    ps_1.7.6           
[55] stringi_1.8.3       gtable_0.3.4        later_1.3.2        
[58] munsell_0.5.0       pillar_1.9.0        htmltools_0.5.7    
[61] R6_2.5.1            textshaping_0.3.7   rprojroot_2.0.4    
[64] evaluate_0.23       highr_0.10          renv_1.0.3         
[67] httpuv_1.6.14       bslib_0.6.1         Rcpp_1.0.12        
[70] gridExtra_2.3       whisker_0.4.1       xfun_0.41          
[73] fs_1.6.3            getPass_0.2-4       pkgconfig_2.0.3    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
