<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Florian Wuennemann" />


<title>Deep Visual Proteomics - snRNA-seq comparison</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mi_spatialomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="data_processing.html">Data processing</a>
</li>
<li>
  <a href="data_analysis.html">Data analysis</a>
</li>
<li>
  <a href="figures.html">Figures</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Deep Visual Proteomics - snRNA-seq
comparison</h1>
<h4 class="author">Florian Wuennemann</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-03-21
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>mi_spatialomics/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230612code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230612)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230612code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230612)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomFloWuennemispatialomicstreee6213a5ae03dcd38ce227600f6ad3c4c9a7c5fc0targetblanke6213a5a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/FloWuenne/mi_spatialomics/tree/e6213a5ae03dcd38ce227600f6ad3c4c9a7c5fc0" target="_blank">e6213a5</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomFloWuennemispatialomicstreee6213a5ae03dcd38ce227600f6ad3c4c9a7c5fc0targetblanke6213a5a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/FloWuenne/mi_spatialomics/tree/e6213a5ae03dcd38ce227600f6ad3c4c9a7c5fc0" target="_blank">e6213a5</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/deprecated/.DS_Store
    Ignored:    analysis/molecular_cartography_python/.DS_Store
    Ignored:    analysis/seqIF_python/.DS_Store
    Ignored:    analysis/seqIF_python/pixie/.DS_Store
    Ignored:    analysis/seqIF_python/pixie/cell_clustering/
    Ignored:    annotations/.DS_Store
    Ignored:    annotations/SeqIF/.DS_Store
    Ignored:    annotations/molkart/.DS_Store
    Ignored:    annotations/molkart/Figure1_regions/.DS_Store
    Ignored:    annotations/molkart/Supplementary_Figure4_regions/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/140623.calcagno_et_al.seurat_object.rds
    Ignored:    data/Calcagno2022_int_logNorm_annot.h5Seurat
    Ignored:    data/IC_03_IF_CCR2_CD68 cell numbers.xlsx
    Ignored:    data/Traditional_IF_absolute_cell_counts.csv
    Ignored:    data/Traditional_IF_relative_cell_counts.csv
    Ignored:    data/pixie.cell_table_size_normalized_cell_labels.csv
    Ignored:    data/results_cts_100.sqm
    Ignored:    data/seqIF_regions_annotations/
    Ignored:    data/seurat/
    Ignored:    omnipathr-log/
    Ignored:    output/.DS_Store
    Ignored:    output/mol_cart.harmony_object.h5Seurat
    Ignored:    output/molkart/
    Ignored:    output/proteomics/
    Ignored:    output/results_cts.lowres.125.sqm
    Ignored:    output/seqIF/
    Ignored:    pipeline_configs/.DS_Store
    Ignored:    plots/
    Ignored:    references/.DS_Store
    Ignored:    renv/.DS_Store
    Ignored:    renv/library/
    Ignored:    renv/staging/

Untracked files:
    Untracked:  analysis/deprecated/figures.supplementary_figureX.Rmd
    Untracked:  analysis/deprecated/figures.supplementary_figure_X.MistyR.Rmd

Unstaged changes:
    Deleted:    analysis/figures.supplementary_figureX.Rmd
    Deleted:    analysis/figures.supplementary_figure_X.MistyR.Rmd
    Deleted:    analysis/figures.supplementary_figure_X.proteomics_qc.Rmd
    Deleted:    figures/Figure_5.eps
    Deleted:    figures/Figure_5.pdf
    Deleted:    figures/Figure_5.png
    Deleted:    figures/Figure_5.svg
    Deleted:    figures/Supplementary_Figure_1_Molecular_Cartography_ROIs.png
    Deleted:    figures/Supplementary_figure_5.segmentation_metrics.poster.eps
    Modified:   figures/Supplementary_figure_X.proteomics.eps
    Modified:   figures/Supplementary_figure_X.proteomics.png
    Modified:   results_cts.lowres.125.sqm

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/proteomics.scRNAseq_comparison.Rmd</code>) and HTML
(<code>docs/proteomics.scRNAseq_comparison.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/56559c732914d3853c5fb73bef8b60761246d181/analysis/proteomics.scRNAseq_comparison.Rmd" target="_blank">56559c7</a>
</td>
<td>
FloWuenne
</td>
<td>
2024-03-21
</td>
<td>
Cleaned up repository.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/f35df9defd28d524467c60e2b4e3c7c02934dcb2/analysis/proteomics.scRNAseq_comparison.Rmd" target="_blank">f35df9d</a>
</td>
<td>
FloWuenne
</td>
<td>
2024-03-02
</td>
<td>
Updated figures for proteomic results.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/FloWuenne/mi_spatialomics/2dcd17804417f9c9b7879f97bfe4b1fd7b5d5510/docs/proteomics.scRNAseq_comparison.html" target="_blank">2dcd178</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-12-06
</td>
<td>
wflow_publish("*")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/5dee03d1f3791ef02f4b04f3ffcef6dca79a876b/analysis/proteomics.scRNAseq_comparison.Rmd" target="_blank">5dee03d</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-09-04
</td>
<td>
Latest code update.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/FloWuenne/mi_spatialomics/5dee03d1f3791ef02f4b04f3ffcef6dca79a876b/docs/proteomics.scRNAseq_comparison.html" target="_blank">5dee03d</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-09-04
</td>
<td>
Latest code update.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/FloWuenne/mi_spatialomics/67e546d5865fdd50e7218ad805ecec53dce079f7/docs/proteomics.scRNAseq_comparison.html" target="_blank">67e546d</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-07-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/ed31d8122236ad8d000508580dcbeb6c25acc4fd/analysis/proteomics.scRNAseq_comparison.Rmd" target="_blank">ed31d81</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-07-02
</td>
<td>
Finalized proteomics analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/FloWuenne/mi_spatialomics/ed31d8122236ad8d000508580dcbeb6c25acc4fd/docs/proteomics.scRNAseq_comparison.html" target="_blank">ed31d81</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-07-02
</td>
<td>
Finalized proteomics analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/FloWuenne/mi_spatialomics/c1395e6d499b82aeae23bc70d4d7622466876c31/docs/proteomics.scRNAseq_comparison.html" target="_blank">c1395e6</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-06-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/236130c306124d35e3cf074cec0c78c955623855/analysis/proteomics.scRNAseq_comparison.Rmd" target="_blank">236130c</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-06-20
</td>
<td>
Updating proteomic analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/FloWuenne/mi_spatialomics/236130c306124d35e3cf074cec0c78c955623855/docs/proteomics.scRNAseq_comparison.html" target="_blank">236130c</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-06-20
</td>
<td>
Updating proteomic analysis.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
</div>
<div id="load-data" class="section level1">
<h1>Load data</h1>
<p>We reprocessed the data from Calcagno et al, using the original
cell-type annotations from the paper and will load this dataset as a
seurat object.</p>
<pre class="r"><code>## Load reprocessed Calcagno et al seurat object
calcagno_et_al &lt;- LoadH5Seurat(&quot;./data/Calcagno2022_int_logNorm_annot.h5Seurat&quot;)</code></pre>
<pre><code>Validating h5Seurat file</code></pre>
<pre><code>Initializing RNA with data</code></pre>
<pre><code>Adding counts for RNA</code></pre>
<pre><code>Adding miscellaneous information for RNA</code></pre>
<pre><code>Initializing integrated with data</code></pre>
<pre><code>Adding scale.data for integrated</code></pre>
<pre><code>Adding variable feature information for integrated</code></pre>
<pre><code>Adding miscellaneous information for integrated</code></pre>
<pre><code>Adding reduction pca</code></pre>
<pre><code>Adding cell embeddings for pca</code></pre>
<pre><code>Adding feature loadings for pca</code></pre>
<pre><code>Adding miscellaneous information for pca</code></pre>
<pre><code>Adding reduction umap</code></pre>
<pre><code>Adding cell embeddings for umap</code></pre>
<pre><code>Adding miscellaneous information for umap</code></pre>
<pre><code>Adding graph integrated_nn</code></pre>
<pre><code>Adding graph integrated_snn</code></pre>
<pre><code>Adding command information</code></pre>
<pre><code>Adding cell-level metadata</code></pre>
<pre><code>Adding miscellaneous information</code></pre>
<pre><code>Adding tool-specific results</code></pre>
<pre><code>Adding data that was not associated with an assay</code></pre>
<pre><code>Warning: Adding a command log without an assay associated with it</code></pre>
<pre class="r"><code>## Get only control cells for marker calculation
calcagno_et_al_ds &lt;- subset(calcagno_et_al,time == &quot;D0&quot;)
calcagno_et_al_d1 &lt;- subset(calcagno_et_al,time == &quot;D1&quot;)</code></pre>
<p>Let’s check the UMAP embedding from our reprocessed object.</p>
<pre class="r"><code>## UMAP plot
DimPlot(calcagno_et_al,label = TRUE) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/proteomics.scRNAseq_comparison.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/5dee03d1f3791ef02f4b04f3ffcef6dca79a876b/docs/figure/proteomics.scRNAseq_comparison.Rmd/unnamed-chunk-3-1.png" target="_blank">5dee03d</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-09-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/ed31d8122236ad8d000508580dcbeb6c25acc4fd/docs/figure/proteomics.scRNAseq_comparison.Rmd/unnamed-chunk-3-1.png" target="_blank">ed31d81</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-07-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Next, let’s quickly verify, that Endocardial cells are expressing the
proper markers before we compare them to our proteomic data.</p>
<pre class="r"><code>plot_density(calcagno_et_al, features = &quot;Npr3&quot;)</code></pre>
<p><img src="figure/proteomics.scRNAseq_comparison.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/5dee03d1f3791ef02f4b04f3ffcef6dca79a876b/docs/figure/proteomics.scRNAseq_comparison.Rmd/unnamed-chunk-4-1.png" target="_blank">5dee03d</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-09-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/ed31d8122236ad8d000508580dcbeb6c25acc4fd/docs/figure/proteomics.scRNAseq_comparison.Rmd/unnamed-chunk-4-1.png" target="_blank">ed31d81</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-07-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(calcagno_et_al, features = &quot;Npr3&quot;)</code></pre>
<p><img src="figure/proteomics.scRNAseq_comparison.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-2">
Past versions of unnamed-chunk-4-2.png
</button>
</p>
<div id="fig-unnamed-chunk-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/2dcd17804417f9c9b7879f97bfe4b1fd7b5d5510/docs/figure/proteomics.scRNAseq_comparison.Rmd/unnamed-chunk-4-2.png" target="_blank">2dcd178</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-12-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/5dee03d1f3791ef02f4b04f3ffcef6dca79a876b/docs/figure/proteomics.scRNAseq_comparison.Rmd/unnamed-chunk-4-2.png" target="_blank">5dee03d</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-09-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/ed31d8122236ad8d000508580dcbeb6c25acc4fd/docs/figure/proteomics.scRNAseq_comparison.Rmd/unnamed-chunk-4-2.png" target="_blank">ed31d81</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-07-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Expression of the endocardial specific marker Npr3 in this dataset
fits with the original authors annotation, suggesting that we can use
these endocardial single-cell signature to identify endocardial specific
genes in our proteomics data.</p>
</div>
<div id="analyze-cell-type-specific-proteins-in-proteomic-data"
class="section level1">
<h1>Analyze cell-type specific proteins in proteomic data</h1>
<p>We will use the snRNAseq data to identify proteins likely
differentially expressed in endocardial cells. For this, we will first
identify genes specifically expressed in endocardial cells.</p>
<div
id="correlate-pseudobulk-snrna-seq-expression-in-endocardial-cells-with-proteomic-measurements"
class="section level2">
<h2>Correlate pseudobulk snRNA-seq expression in endocardial cells with
proteomic measurements</h2>
<p>Let’s load the proteomic data now:</p>
<pre class="r"><code>limma_res &lt;- fread(&quot;./output/proteomics/proteomics.limma.full_statistics.tsv&quot;)

## Extract statistics for different contrasts
miiz_vs_control_signature &lt;- subset(limma_res,analysis == &quot;MI_IZ_vs_control&quot;)
miiz_vs_remote_signature &lt;- subset(limma_res,analysis == &quot;MI_IZ_vs_MI_remote&quot;)

## Load the normalized protein matrix as well
protein_mat &lt;- fread(file = &quot;./output/proteomics/proteomics.vsn_norm_proteins.tsv&quot;)
protein_mat_avg &lt;- protein_mat %&gt;%
  mutate(avg_control=rowMeans(.[ , c(&quot;control_r1&quot;,&quot;control_r2&quot;,&quot;control_r3&quot;)], na.rm=TRUE)) %&gt;%
  mutate(avg_MI_IZ=rowMeans(.[ , c(&quot;MI_IZ_r1&quot;,&quot;MI_IZ_r2&quot;,&quot;MI_IZ_r3&quot;,&quot;MI_IZ_r4&quot;)], na.rm=TRUE)) %&gt;%
  mutate(avg_MI_remote=rowMeans(.[ , c(&quot;MI_remote_r1&quot;,&quot;MI_remote_r2&quot;,&quot;MI_remote_r3&quot;,&quot;MI_remote_r4&quot;)], na.rm=TRUE)) %&gt;%
  dplyr::select(gene,avg_control,avg_MI_IZ,avg_MI_remote)</code></pre>
<pre class="r"><code>## Calculate pseudobulk expression profiles for endocardial cells
endocard_seurat &lt;- subset(calcagno_et_al, level_2 == &quot;Endocardial&quot;)
sn_endo_bulk &lt;- AverageExpression(endocard_seurat, group.by = c(&quot;time&quot;),slot= &quot;data&quot;)</code></pre>
<pre><code>Warning: The `slot` argument of `AverageExpression()` is deprecated as of Seurat 5.0.0.
ℹ Please use the `layer` argument instead.
ℹ The deprecated feature was likely used in the Seurat package.
  Please report the issue at &lt;https://github.com/satijalab/seurat/issues&gt;.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<pre><code>As of Seurat v5, we recommend using AggregateExpression to perform pseudo-bulk analysis.
First group.by variable `time` starts with a number, appending `g` to ensure valid variable names
This message is displayed once per session.</code></pre>
<pre class="r"><code>sn_endo_bulk_df &lt;- as.data.frame(sn_endo_bulk$RNA)
sn_endo_bulk_df$gene &lt;- rownames(sn_endo_bulk_df)</code></pre>
<pre class="r"><code>## Merge average protein expression values with average RNA expression
rna_protein_avg &lt;- left_join(protein_mat_avg,sn_endo_bulk_df, by = &quot;gene&quot;) %&gt;%
  drop_na()

corrplot_rna_protein &lt;- ggplot(rna_protein_avg,aes(avg_control,D0, label = gene)) +
  geom_point() +
  geom_point(data = subset(rna_protein_avg, gene == &quot;Vwf&quot;),color = &quot;red&quot;, size =3) +
  labs(x = &quot;Average protein expression (Control)&quot;,
       y = &quot;Average snRNA-seq expression Control)&quot;)
corrplot_rna_protein</code></pre>
<p><img src="figure/proteomics.scRNAseq_comparison.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/2dcd17804417f9c9b7879f97bfe4b1fd7b5d5510/docs/figure/proteomics.scRNAseq_comparison.Rmd/unnamed-chunk-7-1.png" target="_blank">2dcd178</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-12-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/5dee03d1f3791ef02f4b04f3ffcef6dca79a876b/docs/figure/proteomics.scRNAseq_comparison.Rmd/unnamed-chunk-7-1.png" target="_blank">5dee03d</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-09-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/ed31d8122236ad8d000508580dcbeb6c25acc4fd/docs/figure/proteomics.scRNAseq_comparison.Rmd/unnamed-chunk-7-1.png" target="_blank">ed31d81</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-07-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="calculate-differentially-expressed-genes-in-endocardial-cells-in-snrna-seq"
class="section level2">
<h2>Calculate differentially expressed genes in endocardial cells in
snRNA-seq</h2>
<pre class="r"><code>calcagno_et_al$cell_type_time &lt;- paste(calcagno_et_al$level_2, calcagno_et_al$time,
    sep = &quot;_&quot;)
Idents(calcagno_et_al) &lt;- &quot;cell_type_time&quot;

endocard_de &lt;- FindMarkers(calcagno_et_al, 
                           ident.1 = &quot;Endocardial_D1&quot;, 
                           ident.2 = &quot;Endocardial_D0&quot;,
                           min.diff.pct = 0.1,
                           logfc.threshold = 0,
                           verbose = FALSE)

colnames(endocard_de) &lt;- gsub(&quot;\\.&quot;,&quot;_&quot;,colnames(endocard_de))
endocard_de &lt;- endocard_de %&gt;% 
  mutate(&quot;gene&quot; = rownames(endocard_de)) %&gt;%
  mutate(&quot;pct_ratio&quot; = pct_2 /pct_1,
         &quot;pct_diff&quot; = pct_2 -pct_1) %&gt;%
  arrange(desc(avg_log2FC))</code></pre>
</div>
<div id="get-endocardial-specific-genes" class="section level2">
<h2>Get endocardial specific genes</h2>
<p>To get an estimate of which genes are most specifically expressed in
endocardialc ells in the snRNA-seq data, we will use the FindMarkers
function from Seurat to get log-fold changes and p-values for the
comparison of endocardial cells at day 0 to all other cells.</p>
<pre class="r"><code>endo_marker &lt;- FindMarkers(calcagno_et_al,ident.1 = &quot;Endocardial_D0&quot;,
                           only.pos = TRUE)</code></pre>
<pre class="r"><code>endo_marker$gene &lt;- rownames(endo_marker)
endo_marker &lt;- endo_marker %&gt;%
  mutate(&quot;pct_diff&quot; = pct.1 - pct.2) %&gt;% # Only 
  mutate(&quot;pct_ratio&quot; = pct.1 / pct.2) %&gt;%
  subset(pct.2 &lt; 0.1)                         </code></pre>
</div>
<div
id="compare-endocardial-specific-genes-with-differentially-expressed-proteins-deps"
class="section level2">
<h2>Compare endocardial specific genes with differentially expressed
proteins (DEPs)</h2>
<pre class="r"><code>merged_protein_rna &lt;- left_join(endo_marker,miiz_vs_remote_signature, by = &quot;gene&quot;)

merged_protein_rna &lt;- merged_protein_rna %&gt;%
  mutate(&quot;label_gene&quot; = if_else(gene %in% c(&quot;Vwf&quot;,&quot;Npr3&quot;),gene,&quot;&quot;))

endo_proteomic_corr &lt;- ggplot(merged_protein_rna,aes(avg_log2FC,logFC,
                              label = label_gene)) +
  geom_point(data =subset(merged_protein_rna,gene != &quot;Vwf&quot;), size =3, fill = &quot;darkgrey&quot;, pch = 21) +
  geom_point(data = subset(merged_protein_rna,gene == &quot;Vwf&quot;),size = 4, fill = &quot;red&quot;, pch = 21) +
  geom_point(data = subset(merged_protein_rna,gene == &quot;Npr3&quot;),size = 4, fill = &quot;purple&quot;, pch = 21) +
  geom_label_repel() +
  labs(x = &quot;Specificity for endocardial cells (snRNA-seq)&quot;,
       y = &quot;Log-fold change MI_IZ vs control(proteomics)&quot;)

endo_proteomic_corr</code></pre>
<pre><code>Warning: Removed 3236 rows containing missing values (`geom_point()`).</code></pre>
<pre><code>Warning: Removed 3236 rows containing missing values (`geom_label_repel()`).</code></pre>
<p><img src="figure/proteomics.scRNAseq_comparison.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>write.table(merged_protein_rna,
            file = &quot;./output/proteomics/proteomics.snRNAseq_comp.tsv&quot;,
            sep = &quot;\t&quot;,
            col.names = TRUE,
            row.names = FALSE,
            quote = FALSE)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.1.2

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] RColorBrewer_1.1-3    ggsci_3.0.0           cowplot_1.1.2        
 [4] SeuratDisk_0.0.0.9021 plotly_4.10.4         ggrepel_0.9.5        
 [7] data.table_1.14.10    Nebulosa_1.12.0       patchwork_1.2.0      
[10] Libra_1.7             nnls_1.5              here_1.0.1           
[13] Seurat_5.0.1          SeuratObject_5.0.1    sp_2.1-2             
[16] lubridate_1.9.3       forcats_1.0.0         stringr_1.5.1        
[19] dplyr_1.1.4           purrr_1.0.2           readr_2.1.5          
[22] tidyr_1.3.0           tibble_3.2.1          ggplot2_3.4.4        
[25] tidyverse_2.0.0       workflowr_1.7.1      

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.21            splines_4.3.1              
  [3] later_1.3.2                 bitops_1.0-7               
  [5] polyclip_1.10-6             fastDummies_1.7.3          
  [7] lifecycle_1.0.4             rprojroot_2.0.4            
  [9] hdf5r_1.3.8                 globals_0.16.2             
 [11] processx_3.8.3              lattice_0.22-5             
 [13] MASS_7.3-60.0.1             magrittr_2.0.3             
 [15] limma_3.58.1                sass_0.4.8                 
 [17] rmarkdown_2.25              jquerylib_0.1.4            
 [19] yaml_2.3.8                  httpuv_1.6.14              
 [21] sctransform_0.4.1           spam_2.10-0                
 [23] spatstat.sparse_3.0-3       reticulate_1.34.0          
 [25] pbapply_1.7-2               abind_1.4-5                
 [27] zlibbioc_1.48.0             Rtsne_0.17                 
 [29] GenomicRanges_1.54.1        presto_1.0.0               
 [31] BiocGenerics_0.48.1         RCurl_1.98-1.14            
 [33] pracma_2.4.4                git2r_0.33.0               
 [35] GenomeInfoDbData_1.2.11     IRanges_2.36.0             
 [37] S4Vectors_0.40.2            irlba_2.3.5.1              
 [39] listenv_0.9.0               spatstat.utils_3.0-4       
 [41] goftest_1.2-3               RSpectra_0.16-1            
 [43] spatstat.random_3.2-2       fitdistrplus_1.1-11        
 [45] parallelly_1.36.0           leiden_0.4.3.1             
 [47] codetools_0.2-19            DelayedArray_0.28.0        
 [49] tidyselect_1.2.0            farver_2.1.1               
 [51] matrixStats_1.2.0           stats4_4.3.1               
 [53] spatstat.explore_3.2-5      jsonlite_1.8.8             
 [55] ks_1.14.2                   ellipsis_0.3.2             
 [57] progressr_0.14.0            ggridges_0.5.5             
 [59] survival_3.5-7              tools_4.3.1                
 [61] ica_1.0-3                   Rcpp_1.0.12                
 [63] glue_1.7.0                  gridExtra_2.3              
 [65] SparseArray_1.2.3           xfun_0.41                  
 [67] MatrixGenerics_1.14.0       GenomeInfoDb_1.38.5        
 [69] withr_2.5.2                 BiocManager_1.30.22        
 [71] fastmap_1.1.1               fansi_1.0.6                
 [73] callr_3.7.3                 digest_0.6.34              
 [75] timechange_0.2.0            R6_2.5.1                   
 [77] mime_0.12                   colorspace_2.1-0           
 [79] scattermore_1.2             tensor_1.5                 
 [81] spatstat.data_3.0-3         utf8_1.2.4                 
 [83] generics_0.1.3              renv_1.0.3                 
 [85] httr_1.4.7                  htmlwidgets_1.6.4          
 [87] S4Arrays_1.2.0              whisker_0.4.1              
 [89] uwot_0.1.16                 pkgconfig_2.0.3            
 [91] gtable_0.3.4                lmtest_0.9-40              
 [93] SingleCellExperiment_1.24.0 XVector_0.42.0             
 [95] htmltools_0.5.7             dotCall64_1.1-1            
 [97] scales_1.3.0                Biobase_2.62.0             
 [99] png_0.1-8                   knitr_1.45                 
[101] rstudioapi_0.15.0           tzdb_0.4.0                 
[103] reshape2_1.4.4              nlme_3.1-164               
[105] cachem_1.0.8                zoo_1.8-12                 
[107] KernSmooth_2.23-22          vipor_0.4.7                
[109] parallel_4.3.1              miniUI_0.1.1.1             
[111] ggrastr_1.0.2               pillar_1.9.0               
[113] grid_4.3.1                  vctrs_0.6.5                
[115] RANN_2.6.1                  promises_1.2.1             
[117] xtable_1.8-4                cluster_2.1.6              
[119] beeswarm_0.4.0              evaluate_0.23              
[121] mvtnorm_1.2-4               cli_3.6.2                  
[123] compiler_4.3.1              rlang_1.1.3                
[125] crayon_1.5.2                future.apply_1.11.1        
[127] labeling_0.4.3              mclust_6.0.1               
[129] ps_1.7.6                    ggbeeswarm_0.7.2           
[131] getPass_0.2-4               plyr_1.8.9                 
[133] fs_1.6.3                    stringi_1.8.3              
[135] viridisLite_0.4.2           deldir_2.0-2               
[137] munsell_0.5.0               lazyeval_0.2.2             
[139] spatstat.geom_3.2-7         Matrix_1.6-5               
[141] RcppHNSW_0.5.0              hms_1.1.3                  
[143] bit64_4.0.5                 future_1.33.1              
[145] statmod_1.5.0               shiny_1.8.0                
[147] highr_0.10                  SummarizedExperiment_1.32.0
[149] ROCR_1.0-11                 igraph_1.6.0               
[151] bslib_0.6.1                 bit_4.0.5                  </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
