<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="FloWuenne" />

<meta name="date" content="2023-08-21" />

<title>lunaphore.heatmaps_pixie.figure3</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mi_spatialomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="data_processing.html">Data processing</a>
</li>
<li>
  <a href="data_analysis.html">Data analysis</a>
</li>
<li>
  <a href="figures.html">Figures</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">lunaphore.heatmaps_pixie.figure3</h1>
<h4 class="author">FloWuenne</h4>
<h4 class="date">2023-08-21</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-08-24
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>mi_spatialomics/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230612code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230612)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230612code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230612)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomFloWuennemispatialomicstreee8204f0f7affd3e57c428cc8708d1de9623632d1targetblanke8204f0a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/FloWuenne/mi_spatialomics/tree/e8204f0f7affd3e57c428cc8708d1de9623632d1" target="_blank">e8204f0</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomFloWuennemispatialomicstreee8204f0f7affd3e57c428cc8708d1de9623632d1targetblanke8204f0a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/FloWuenne/mi_spatialomics/tree/e8204f0f7affd3e57c428cc8708d1de9623632d1" target="_blank">e8204f0</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/deprecated/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/140623.calcagno_et_al.seurat_object.rds
    Ignored:    data/molkart_tissue_regions_rois/.DS_Store
    Ignored:    figures/.DS_Store
    Ignored:    omnipathr-log/
    Ignored:    output/.DS_Store
    Ignored:    output/figure3.pixie_cell_cluster_heatmap.png
    Ignored:    output/figure3.pixie_pixel_cluster_heatmap.png
    Ignored:    output/figure3.pixie_pixel_cluster_heatmap.tiff
    Ignored:    output/harmony.molkart.h5Seurat
    Ignored:    output/lunaphore_images/
    Ignored:    output/mol_cart/
    Ignored:    output/molkart_cell_types/
    Ignored:    output/proteomics/
    Ignored:    output/seqIF/
    Ignored:    plots/.DS_Store
    Ignored:    plots/Figure1.umap_plot.pdf
    Ignored:    references/.DS_Store
    Ignored:    renv/.DS_Store
    Ignored:    renv/library/
    Ignored:    renv/staging/

Untracked files:
    Untracked:  analysis/__pycache__/
    Untracked:  analysis/deprecated/molkart.quantify_cells_in_regions.ipynb
    Untracked:  analysis/deprecated/napari_points.ipynb
    Untracked:  analysis/deprecated/roi.csv
    Untracked:  analysis/deprecated/roi2.csv
    Untracked:  analysis/deprecated/test_construct_spatialdata.ipynb
    Untracked:  analysis/figures.deep_visual_proteomics.Rmd
    Untracked:  analysis/mol_cart.QC_spots.Rmd
    Untracked:  analysis/mol_cart.molkart.Figure1.Rmd
    Untracked:  analysis/mol_cart.molkart.process_quantifications_seurat.Rmd
    Untracked:  analysis/molecular_cartography_python/
    Untracked:  analysis/seqIF.heatmaps_pixie.figure3.Rmd
    Untracked:  analysis/seqIF_python/
    Untracked:  analysis/spatialMI_functions.py
    Untracked:  data/Calcagno2022_int_logNorm_annot.h5Seurat
    Untracked:  data/pixie.cell_table_size_normalized_cell_labels.csv
    Untracked:  plots/Figure3.cell_types_overtimes.pdf
    Untracked:  plots/Figure3.pixel_clusters_overtimes.pdf
    Untracked:  references/mol_cart.heart_regions/

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/data_analysis.Rmd
    Modified:   analysis/data_processing.Rmd
    Modified:   analysis/figures.Rmd
    Deleted:    analysis/figures.figure_5.Rmd
    Modified:   analysis/figures.supplementary_figure_X.proteomics_qc.Rmd
    Deleted:    analysis/molkart.Figure1.Rmd
    Deleted:    analysis/molkart.QC_spots.Rmd
    Deleted:    analysis/molkart.process_quantifications_seurat.Rmd
    Modified:   analysis/proteomics.bulk_de_analysis.Rmd
    Modified:   analysis/proteomics.filter_proteomic_data.Rmd
    Modified:   analysis/proteomics.pathway_enrichment_analysis.Rmd
    Modified:   analysis/proteomics.scRNAseq_comparison.Rmd
    Deleted:    analysis/python/lunaphore.figure_3.create_pixie_images.ipynb
    Deleted:    analysis/python/lunaphore.figure_3.pixie_heatmaps.ipynb
    Deleted:    analysis/python/molkart.count_spots_on_tissue.ipynb
    Deleted:    analysis/python/molkart.plot_MC_spots.ipynb
    Deleted:    analysis/python/molkart.plot_spots_figure1.ipynb
    Deleted:    analysis/python/molkart.quantify_cells_in_regions.ipynb
    Deleted:    analysis/python/napari_points.ipynb
    Deleted:    analysis/python/roi.csv
    Deleted:    analysis/python/roi2.csv
    Deleted:    analysis/python/spatialMI_functions.py
    Deleted:    analysis/python/test_construct_spatialdata.ipynb
    Deleted:    data/molkart.spots_per_tissue.tsv
    Modified:   figures/Figure_5.eps
    Modified:   figures/Figure_5.pdf
    Modified:   figures/Figure_5.png
    Modified:   figures/Figure_5.svg
    Modified:   output/molcart.misty_celltype_table.tsv
    Deleted:    output/molkart_segmentation_images/sample_control_r1_s1.DAPI_WGA.crop.png
    Deleted:    output/molkart_segmentation_images/sample_control_r1_s1.DAPI_WGA.crop.scale.png
    Deleted:    output/molkart_segmentation_images/sample_control_r1_s1.DAPI_WGA.roi.tif
    Deleted:    output/molkart_segmentation_images/sample_control_r1_s1.DAPI_WGA.tif
    Deleted:    output/molkart_segmentation_images/sample_control_r1_s1.DAPI_WGA_roi.crop.png
    Deleted:    output/molkart_segmentation_images/sample_control_r1_s1.DAPI_WGA_roi.crop.scale.png
    Deleted:    output/molkart_segmentation_images/sample_control_r1_s1.cellpose_full_image.outline.tif
    Deleted:    output/molkart_segmentation_images/sample_control_r1_s1.cellpose_full_image.roi.outline.tif
    Deleted:    output/molkart_segmentation_images/sample_control_r1_s1.cellpose_full_image.roi.tif
    Deleted:    output/molkart_segmentation_images/sample_control_r1_s1.cellpose_full_image.tif
    Deleted:    output/molkart_segmentation_images/sample_control_r1_s1.cellpose_mask.crop.png
    Deleted:    output/molkart_segmentation_images/sample_control_r1_s1.cellpose_mask.crop.scale.png
    Deleted:    output/molkart_segmentation_images/sample_control_r1_s1.cellpose_mask_roi.crop.png
    Deleted:    output/molkart_segmentation_images/sample_control_r1_s1.cellpose_mask_roi.crop.scale.png
    Deleted:    output/proteomics.filt_imputed_proteins.tsv
    Deleted:    output/proteomics.filtered_proteins.tsv
    Deleted:    output/proteomics.pca_res.rds
    Deleted:    output/proteomics.protein_missing_stats.tsv
    Deleted:    output/proteomics.vsn_norm_proteins.tsv
    Modified:   plots/Figure1.dotplot.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(pheatmap)
library(data.table)
library(viridis)</code></pre>
<pre><code>Loading required package: viridisLite</code></pre>
<pre class="r"><code>library(RColorBrewer)
library(tidyverse)</code></pre>
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.2     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.2     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.1     </code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::between()     masks data.table::between()
✖ dplyr::filter()      masks stats::filter()
✖ dplyr::first()       masks data.table::first()
✖ lubridate::hour()    masks data.table::hour()
✖ lubridate::isoweek() masks data.table::isoweek()
✖ dplyr::lag()         masks stats::lag()
✖ dplyr::last()        masks data.table::last()
✖ lubridate::mday()    masks data.table::mday()
✖ lubridate::minute()  masks data.table::minute()
✖ lubridate::month()   masks data.table::month()
✖ lubridate::quarter() masks data.table::quarter()
✖ lubridate::second()  masks data.table::second()
✖ purrr::transpose()   masks data.table::transpose()
✖ lubridate::wday()    masks data.table::wday()
✖ lubridate::week()    masks data.table::week()
✖ lubridate::yday()    masks data.table::yday()
✖ lubridate::year()    masks data.table::year()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<pre class="r"><code>library(pals)</code></pre>
<pre><code>
Attaching package: &#39;pals&#39;

The following objects are masked from &#39;package:viridis&#39;:

    cividis, inferno, magma, plasma, turbo, viridis

The following objects are masked from &#39;package:viridisLite&#39;:

    cividis, inferno, magma, plasma, turbo, viridis</code></pre>
<pre class="r"><code>library(vroom)</code></pre>
<pre><code>
Attaching package: &#39;vroom&#39;

The following objects are masked from &#39;package:readr&#39;:

    as.col_spec, col_character, col_date, col_datetime, col_double,
    col_factor, col_guess, col_integer, col_logical, col_number,
    col_skip, col_time, cols, cols_condense, cols_only, date_names,
    date_names_lang, date_names_langs, default_locale, fwf_cols,
    fwf_empty, fwf_positions, fwf_widths, locale, output_column,
    problems, spec</code></pre>
<pre class="r"><code>pixel_map_color &lt;- c(&quot;#0173B2&quot;, &quot;#DE8F05&quot;, &quot;#029E73&quot;, &quot;#D55E00&quot;, &quot;#CC78BC&quot;,
                &quot;#CA9161&quot;, &quot;#FBAFE4&quot;, &quot;#949494&quot;, &quot;#ECE133&quot;, &quot;#56B4E9&quot;)
cell_cluster_color &lt;- glasbey()

source(&quot;./code/functions.R&quot;)</code></pre>
<pre><code>
Attaching package: &#39;cowplot&#39;

The following object is masked from &#39;package:lubridate&#39;:

    stamp

here() starts at /Users/florian_wuennemann/1_Projects/MI_project/mi_spatialomics</code></pre>
<div id="pixel-cluster" class="section level1">
<h1>Pixel cluster</h1>
<div id="pixel-cluster-maps" class="section level2">
<h2>Pixel cluster maps</h2>
<pre class="r"><code>avg_pixel_cluster &lt;- fread(&quot;/Users/florian_wuennemann/1_Projects/MI_project/Lunaphore/pixie/masked_subset/subset_0.05_wseg/pixel_channel_avg_meta_cluster.csv&quot;)
avg_pixel_cluster &lt;- avg_pixel_cluster %&gt;%
  subset(pixel_meta_cluster_rename != &quot;background&quot;)</code></pre>
<pre class="r"><code>mat_rownames &lt;- avg_pixel_cluster$pixel_meta_cluster_rename
mat_rownames &lt;- gsub(&quot;_&quot;,&quot;+ &quot;,mat_rownames)
mat_rownames &lt;- paste(mat_rownames,&quot;pixels&quot;, sep = &quot; &quot;)
mat_dat &lt;- avg_pixel_cluster %&gt;%
  dplyr::select(-c(pixel_meta_cluster,count,pixel_meta_cluster_rename))</code></pre>
<pre class="r"><code>cap = 3 #hierarchical clustering cap
hclust_coln = &quot;pixel_meta_cluster_rename&quot;
rwb_cols = colorRampPalette(c(&quot;royalblue4&quot;,&quot;white&quot;,&quot;red4&quot;))(99)

mat_dat = scale(mat_dat)
mat_dat = pmin(mat_dat, cap)
rownames(mat_dat) &lt;- mat_rownames

# Determine breaks
range = max(abs(mat_dat))
breaks = seq(-range, range, length.out=100)

mat_col = data.frame(pixel_cluster = as.factor(mat_rownames))
rownames(mat_col) &lt;- mat_rownames
mat_colors = pixel_map_color[1:length(mat_rownames)]
names(mat_colors) = mat_rownames
mat_colors = list(pixel_cluster = mat_colors)

# Make heatmap
pheatmap(mat_dat,
         color = rwb_cols,
         border_color = &quot;black&quot;,
         breaks = breaks,
         cluster_rows = TRUE,
         cluster_cols = TRUE,
         treeheight_col = 25,
         treeheight_row = 25,
         #treeheight_col = 0,
         show_rownames = TRUE,
         annotation_row = mat_col,
         annotation_colors = mat_colors,
         annotation_names_row = FALSE,
         annotation_legend = FALSE,
         legend = TRUE,
         #legend_breaks = c(-3,-2,-1,0,1,2,3),
         #legend_labels = c(&quot;-3&quot;,&quot;-2&quot;,&quot;-1&quot;,&quot;0&quot;,&quot;1&quot;,&quot;2&quot;,&quot;3&quot;),
         main = &quot;&quot;,
         filename = &quot;./output/figure3.pixie_pixel_cluster_heatmap.png&quot;,
         fontsize = 20,
         width = 8,
         height = 6)
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
</div>
<div id="pixel-cluster-changes-over-time" class="section level2">
<h2>Pixel cluster changes over time</h2>
<pre class="r"><code>## read in pixel cluster counts
pixel_counts = fread(&quot;/Users/florian_wuennemann/1_Projects/MI_project/Lunaphore/pixie/masked_subset/subset_0.05_wseg/pixel_counts.all_samples.csv&quot;)</code></pre>
<pre class="r"><code>pixel_cluster_counts_stats &lt;- pixel_counts %&gt;%
  subset(Pixel_cluster != &quot;background&quot;) %&gt;%
  separate(&quot;Sample_ID&quot;, into = c(&quot;time&quot;,&quot;sample&quot;)) %&gt;%
  group_by(time,Pixel_cluster) %&gt;%
  summarise(&quot;n_pixel&quot; = sum(Count)) %&gt;%
  mutate(&quot;percent&quot; = n_pixel / sum(n_pixel)) %&gt;%
  ungroup()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;time&#39;. You can override using the
`.groups` argument.</code></pre>
<pre class="r"><code>pixel_cluster_counts_stats$time &lt;-factor(pixel_cluster_counts_stats$time,
                              levels = c(&quot;Control&quot;,&quot;4h&quot;,&quot;24h&quot;,&quot;48h&quot;))

pixel_cluster_counts_stats$time_cont &lt;- as.numeric(pixel_cluster_counts_stats$time)</code></pre>
<pre class="r"><code>#ggplot(cells_over_time, aes(x=time, y=percent, fill=cell_meta_cluster)) +
  #geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;,color = &quot;black&quot;)

pixel_number_distribution &lt;- ggplot(pixel_cluster_counts_stats, 
                                    aes(x=time_cont, y=percent)) +
  geom_area(aes(fill = Pixel_cluster), color = &quot;black&quot;) +
  theme(legend.position = &quot;none&quot;,
        axis.line = element_blank()) +
  scale_fill_manual(values = pixel_map_color) +
  scale_x_discrete(expand = c(0,0.1),
                   name =&quot;Time&quot;, 
                   limits=c(&quot;Control&quot;,&quot;4h&quot;,&quot;24h&quot;,&quot;48h&quot;)) +
  labs(y = &quot;% cells&quot;)

pixel_number_distribution</code></pre>
<p><img src="figure/seqIF.heatmaps_pixie.figure3.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>save_plot(pixel_number_distribution,
          file = &quot;./plots/Figure3.pixel_clusters_overtimes.pdf&quot;,
          base_height = 3.5,
          base_asp = 1)</code></pre>
</div>
</div>
<div id="cell-cluster-map" class="section level1">
<h1>Cell cluster map</h1>
<div id="cell-cluster-tissue-maps" class="section level2">
<h2>Cell cluster tissue maps</h2>
<pre class="r"><code>avg_cell_cluster &lt;- fread(&quot;/Users/florian_wuennemann/1_Projects/MI_project/Lunaphore/pixie/masked_subset/cell_masks_0.05/cell_meta_cluster_channel_avg.csv&quot;)
avg_cell_cluster &lt;- avg_cell_cluster %&gt;%
  subset(cell_meta_cluster_rename != &quot;background&quot;)</code></pre>
<pre class="r"><code>mat_rownames &lt;- avg_cell_cluster$cell_meta_cluster_rename
mat_rownames &lt;- gsub(&quot;_&quot;,&quot;+ &quot;,mat_rownames)
mat_dat &lt;- avg_cell_cluster %&gt;%
  dplyr::select(-c(cell_meta_cluster,cell_meta_cluster_rename))</code></pre>
<pre class="r"><code>cap = 3 #hierarchical clustering cap
hclust_coln = &quot;pixel_meta_cluster_rename&quot;
rwb_cols = colorRampPalette(c(&quot;royalblue4&quot;,&quot;white&quot;,&quot;red4&quot;))(99)

mat_dat = scale(mat_dat)
mat_dat = pmin(mat_dat, cap)
rownames(mat_dat) &lt;- mat_rownames

# Determine breaks
range = max(abs(mat_dat))
breaks = seq(-range, range, length.out=100)

mat_col = data.frame(pixel_cluster = as.factor(mat_rownames))
rownames(mat_col) &lt;- mat_rownames
mat_colors = cell_cluster_color[1:length(mat_rownames)]
names(mat_colors) = mat_rownames
mat_colors = list(pixel_cluster = mat_colors)

# Make heatmap
pheatmap(mat_dat,
         color = rwb_cols,
         border_color = &quot;black&quot;,
         breaks = breaks,
         cluster_rows = TRUE,
         cluster_cols = TRUE,
         treeheight_col = 25,
         treeheight_row = 25,
         #treeheight_col = 0,
         show_rownames = TRUE,
         annotation_row = mat_col,
         annotation_colors = mat_colors,
         annotation_names_row = FALSE,
         annotation_legend = FALSE,
         legend = TRUE,
         #legend_breaks = c(-3,-2,-1,0,1,2,3),
         #legend_labels = c(&quot;-3&quot;,&quot;-2&quot;,&quot;-1&quot;,&quot;0&quot;,&quot;1&quot;,&quot;2&quot;,&quot;3&quot;),
         main = &quot;&quot;,
         filename = &quot;./output/figure3.pixie_cell_cluster_heatmap.png&quot;,
         fontsize = 20,
         width = 8,
         height = 6)
dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
</div>
<div id="cell-cluster-changes-over-time" class="section level2">
<h2>Cell cluster changes over time</h2>
<pre class="r"><code>cells_over_time &lt;- fread(&quot;./data/pixie.cell_table_size_normalized_cell_labels.csv&quot;)
cells_over_time &lt;-  cells_over_time %&gt;%
  mutate(&quot;cell_type_labels&quot; = if_else(grepl(&quot;Myeloid&quot;,cell_meta_cluster),&quot;Myeloid cells&quot;,cell_meta_cluster))

cells_over_time &lt;- cells_over_time %&gt;%
  subset(cell_type_labels != &quot;background&quot;) %&gt;%
  separate(&quot;fov&quot;, into = c(&quot;time&quot;,&quot;sample&quot;)) %&gt;%
  group_by(time,cell_type_labels) %&gt;%
  tally() %&gt;%
  ungroup()</code></pre>
<pre class="r"><code>cells_over_time &lt;- cells_over_time %&gt;%
  group_by(time) %&gt;%
  mutate(&quot;percent&quot; = n / sum(n)) %&gt;%
  ungroup()</code></pre>
<pre class="r"><code>cells_over_time$time &lt;-factor(cells_over_time$time,
                              levels = c(&quot;Control&quot;,&quot;4h&quot;,&quot;24h&quot;,&quot;48h&quot;))

cells_over_time$time_cont &lt;- as.numeric(cells_over_time$time)</code></pre>
<pre class="r"><code>#ggplot(cells_over_time, aes(x=time, y=percent, fill=cell_meta_cluster)) +
  #geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;,color = &quot;black&quot;)
mat_colors = cell_cluster_color[1:length(mat_rownames)]
cell_number_distribution &lt;- ggplot(cells_over_time, aes(x=time_cont, y=percent)) +
  geom_area(aes(fill = cell_type_labels), color = &quot;black&quot;) +
  theme(legend.position = &quot;top&quot;,
        axis.line = element_blank()) +
  scale_fill_manual(values = mat_colors) +
  scale_x_discrete(expand = c(0,0.1),
                   name =&quot;Time&quot;, 
                    limits=c(&quot;Control&quot;,&quot;4h&quot;,&quot;24h&quot;,&quot;48h&quot;)) +
  labs(y = &quot;% cells&quot;)
cell_number_distribution</code></pre>
<p><img src="figure/seqIF.heatmaps_pixie.figure3.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>save_plot(cell_number_distribution,
          file = &quot;./plots/Figure3.cell_types_overtimes.pdf&quot;,
          base_height = 3.5,
          base_asp = 1)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.3 (2023-03-15)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] here_1.0.1         ggsci_3.0.0        cowplot_1.1.1      vroom_1.6.3       
 [5] pals_1.7           lubridate_1.9.2    forcats_1.0.0      stringr_1.5.0     
 [9] dplyr_1.1.2        purrr_1.0.1        readr_2.1.4        tidyr_1.3.0       
[13] tibble_3.2.1       ggplot2_3.4.2      tidyverse_2.0.0    RColorBrewer_1.1-3
[17] viridis_0.6.4      viridisLite_0.4.2  data.table_1.14.8  pheatmap_1.0.12   
[21] workflowr_1.7.0   

loaded via a namespace (and not attached):
 [1] httr_1.4.6            sass_0.4.7            maps_3.4.1           
 [4] bit64_4.0.5           jsonlite_1.8.7        bslib_0.5.0          
 [7] getPass_0.2-2         BiocManager_1.30.21.1 highr_0.10           
[10] renv_1.0.0            yaml_2.3.7            pillar_1.9.0         
[13] glue_1.6.2            digest_0.6.33         promises_1.2.0.1     
[16] colorspace_2.1-0      htmltools_0.5.5       httpuv_1.6.11        
[19] pkgconfig_2.0.3       scales_1.2.1          processx_3.8.2       
[22] whisker_0.4.1         later_1.3.1           tzdb_0.4.0           
[25] timechange_0.2.0      git2r_0.32.0          generics_0.1.3       
[28] farver_2.1.1          cachem_1.0.8          withr_2.5.0          
[31] cli_3.6.1             magrittr_2.0.3        crayon_1.5.2         
[34] evaluate_0.21         ps_1.7.5              fs_1.6.3             
[37] fansi_1.0.4           textshaping_0.3.6     tools_4.2.3          
[40] hms_1.1.3             lifecycle_1.0.3       munsell_0.5.0        
[43] callr_3.7.3           compiler_4.2.3        jquerylib_0.1.4      
[46] systemfonts_1.0.4     rlang_1.1.1           grid_4.2.3           
[49] dichromat_2.0-0.1     rstudioapi_0.15.0     labeling_0.4.2       
[52] rmarkdown_2.23        gtable_0.3.3          R6_2.5.1             
[55] gridExtra_2.3         knitr_1.43            fastmap_1.1.1        
[58] bit_4.0.5             utf8_1.2.3            rprojroot_2.0.3      
[61] ragg_1.2.5            stringi_1.7.12        Rcpp_1.0.11          
[64] vctrs_0.6.3           mapproj_1.2.11        tidyselect_1.2.0     
[67] xfun_0.39            </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
