---
title: "Figure 5"
author: "Florian Wuennemann"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
library(data.table)
library(here)
library(ggrepel)
library(ggbeeswarm)
library(ggsignif)
library(patchwork)
library(ggforce)

source("./code/functions.R")
```


# Load data

```{r}
pca_res <- readRDS("./output/proteomics/proteomics.pca_res.rds")
vsn_mat <- fread("./output/proteomics/proteomics.vsn_norm_proteins.tsv")
limma_res <- fread("./output/proteomics/proteomics.limma.full_statistics.tsv")
mi_pathways <- fread("./output/proteomics/proteomics.pathway_results.MIiz_MIremote.tsv")
```


# Subfigure A


# Subfigure B - Principal component analysis

```{r}
pcs <- as.data.frame(pca_res$x)
pcs$sample <- colnames(vsn_mat[,1:11])
pcs <- pcs %>%
  mutate("group" = if_else(grepl("control",sample),"control",
                          if_else(grepl("MI_IZ",sample),"MI_IZ","MI_remote"))
         )

## Set order of groups
pcs$group <- factor(pcs$group,
                    levels = c("control","MI_remote","MI_IZ"))

## Plot PCs
pca_plot <- ggplot(pcs,aes(PC1,PC2)) +
  geom_point(size = 5,pch = 21,color = "black", aes(fill = group)) +
  ggforce::geom_mark_ellipse(color = "white",aes(fill = group)) +
  expand_limits(y = c(-50, 40),
                x = c(-40,80)) +
  scale_fill_manual(values = proteome_palette,
                    labels = c("Control","MI_remote","MI_IZ")) +
  labs(color = "Group") +
  guides(fill=guide_legend(title="Group")) +
  theme(legend.position = "none")


pca_plot

save_plot(filename = "./plots/Figure_5.pca_plot.pdf",
          plot = pca_plot,
          base_asp = 1,
          base_height = 4)
```


# Subfigure C - Volcano plot: Remote vs control

```{r}
limma_mi_remote <- subset(limma_res,analysis == "MI_IZ_vs_MI_remote")
limma_remote_control <- subset(limma_res,analysis == "MI_remote_vs_control")
limma_mi_control <- subset(limma_res,analysis == "MI_IZ_vs_control")

## Which proteins are differentially expressed in MI vs remote but not in MI vs remote but not in remote vs control?
iz_uniq <- setdiff(subset(limma_mi_remote,adj.P.Val < 0.05)$gene,subset(limma_remote_control,adj.P.Val < 0.05)$gene)
limma_mi_remote_uniq <- subset(limma_mi_remote, gene %in% iz_uniq) %>%
  subset(adj.P.Val < 0.05) %>%
  arrange(desc(logFC))

## Get proteins from Coagulation pathway from pathway analysis results to highlight on volcano plot
mh_gsea_net <- readRDS("./references/mh.all.v2023.1.Mm.symbols.sets.rds")

pathway <- 'HALLMARK_COAGULATION'

df <- mh_gsea_net %>%
  filter(source == pathway) %>%
  arrange(target)

path_de_inter <- sort(intersect(limma_mi_remote$gene,df$target))
```


```{r}
# top_10_proteins <- limma_mi_remote %>%
#   arrange(desc(logFC)) %>%
#   top_n(wt = logFC, 10)
# top_10_proteins <- top_10_proteins$gene
# bottom_10_proteins <- limma_mi_remote %>% arrange(desc(logFC))
# bottom_10_proteins <- tail(bottom_10_proteins,n=10)

manual_labeled_proteins <- c("Thbd","Vwf","Coro1a","Thbs1")

limma_mi_remote <- limma_mi_remote %>%
  # mutate("label_protein" = if_else(gene %in% path_de_inter & adj.P.Val < 0.05 & (logFC > 1.25 | logFC < 0), gene, ""))
  mutate("label_protein" = if_else(gene %in% manual_labeled_proteins,gene,""))

limma_mi_remote$label_protein <- gsub("Vwf","vWF",limma_mi_remote$label_protein)

volc_limma_IZ_remote <- plot_pretty_volcano(limma_mi_remote, 
                                      pt_size = 2,
                                      plot_title = "",
                                      sig_thresh = 0.05,
                                      col_pos_logFC = proteome_palette[['MI_IZ']],
                                      col_neg_logFC = proteome_palette[['MI_remote']]) +
    # geom_point(data = subset(limma_mi_remote, gene %in% path_de_inter),pch = 21, color = "black", size = 4) +
  geom_label_repel(box.padding = 0.5, max.overlaps = Inf) +
  geom_vline(xintercept= 0 , linetype = 2)

## Interactive plotly plot to view genes on points
# plot_ly(data = limma_mi_remote, x = ~logFC, y = ~-log10(adj.P.Val),
#         text = ~paste("Gene: ", gene))

save_plot(filename = "./plots/Figure_5.volcano_plot.pdf",
          plot = volc_limma_IZ_remote,
          base_asp = 1.3,
          base_height = 3.25)
```

```{r}
## Volcano plot for schema
volc_limma_remote_control <- ggplot(data=limma_remote_control, 
                                    aes(x= logFC, y= -log10(pval))) +
    geom_point(size = 3, color = "black")+
  theme(axis.title = element_text(size =20),
        axis.text = element_blank(),
        axis.ticks = element_blank()) +
  geom_vline(xintercept = 0, linetype = 2)

volc_limma_remote_control

save_plot(filename = "./plots/Figure_4.volcano_schema.pdf",
          plot = volc_limma_remote_control,
          base_asp = 1.4,
          base_height = 3)
```


# Subfigure D - Pathway enrichment for MI_IZ vs MI_remote

```{r}
sig_pathways_mi <- subset(mi_pathways,p_value <= 0.05) %>%
  arrange(desc(score)) %>%
  dplyr::select(-statistic,-condition) %>%
  subset(score > 3 | score < -3)

sig_pathways_mi$source <- gsub("HALLMARK_","",sig_pathways_mi$source)

sig_pathways_mi$source <- gsub("_"," ",sig_pathways_mi$source)
path_plot <- ggplot(sig_pathways_mi, aes(x = reorder(source, score), y = score)) + 
    geom_bar(aes(fill = score),color = "black", stat = "identity") +
    scale_fill_gradient2(low = "darkorange", high = "purple",
        mid = "white", midpoint = 0) +
  # scale_fill_viridis(option = "F", direction = 1) +
    theme(axis.title = element_text(face = "bold", size = 20),
          axis.text.x = element_text(hjust = 1, size =20, face= "bold"),
          axis.text.y = element_text(size =16),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          legend.text = element_text(size =20),
          legend.title = element_text(size =20)) +
    xlab("Pathways") +
  coord_flip()

path_plot


save_plot(filename = "./plots/Figure_5.pathway_plot.pdf",
          plot = path_plot,
          base_asp = 2,
          base_height = 4)
```

# Subfigure E - Vwf specificity for Endocardial cells

```{r}
snrna_prot <- fread("./output/proteomics/proteomics.snRNAseq_comp.tsv")
snrna_prot <- snrna_prot %>%
  mutate("label_gene" = if_else(gene %in% c("Cdh11","Thbd","Vcam1"),gene,
                                if_else(gene == "Vwf","vWF",""))) %>%
  subset(pct.1 > 0.05)

endo_proteomic_corr <- ggplot(snrna_prot,aes(avg_log2FC,logFC,
                              label = label_gene)) +
  geom_hline(yintercept = 0, linetype = 2) +
  geom_point(data =subset(snrna_prot,gene != "Vwf"), size =3, fill = "darkgrey", pch = 21) +
  geom_point(data = subset(snrna_prot,gene %in% c("Vwf","Vcam1")),size = 4, fill = "purple", pch = 21) +
  geom_point(data = subset(snrna_prot,gene %in% c("Thbd")),size = 4, fill = "darkorange", pch = 21) +
    geom_point(data = subset(snrna_prot,gene %in% c("Cdh11")),size = 4, fill = "grey20", pch = 21) +
  geom_label_repel(size = 5.5, max.overlaps = 20,force = 3) +
  labs(x = "Specificity endocard. cells (snRNA-seq)",
       y = "Log2 fold-change (Proteomics)") 

endo_proteomic_corr

save_plot(filename = "./plots/Figure_5.vwf_specificity_plot.pdf",
          plot = endo_proteomic_corr,
          base_asp = 1.75,
          base_height = 3.5)
```


# Subfigure F - Vwf is upregulated in MI_IZ

```{r}
source("./code/functions.R")
yaxis_limits <- c(11,17)

vsn_matrix <- fread("./output/proteomics/proteomics.vsn_norm_proteins.tsv")
colnames(vsn_matrix)[1:11] <- paste("s",1:11,colnames(vsn_matrix)[1:11],sep="_")
protein_sub <- vsn_matrix  %>%
    dplyr::select(1:11,gene) %>%
    pivot_longer(1:11,names_to = "sample", values_to = "exp") %>%
    mutate("group" = if_else(grepl("control",sample),"control",
                                   if_else(grepl("MI_IZ",sample),
                                           "MI_IZ","MI_remote"))
    )

protein_sub$group <- gsub("control","Control",protein_sub$group)

protein_sub$group <- factor(protein_sub$group,
                              levels = c("Control","MI_remote","MI_IZ"))


## Barplot with points as alternative.
# goi <- "Vwf"
# vwf_plot_bar <- plot_proteomics_boxplot(norm_table = protein_sub,
#                                     protein = goi,
#                                     style = "bar") +
#   geom_signif(comparisons = list(c("MI_IZ","MI_remote")),
#               tip_length = 0, annotation = "0.0057", y_position = 15.5) +
#   geom_signif(comparisons = list(c("MI_IZ","Control")),
#               tip_length = 0, annotation = "0.0022", y_position = 16.5) +
#    expand_limits(y = c(13, 17)) +
#   theme(axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) +
#   labs(x = "")


## Median plot with points
goi <- "Cdh11"
npr3_plot <- plot_proteomics_boxplot(norm_table = protein_sub,
                                    protein = goi,
                                    style = "mean") +
  theme(axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) +
  labs(x = "") +
  ylim(yaxis_limits) +
  labs(y = "") +
  scale_x_discrete(labels=c("Control" = "Control", 
                            "MI_remote" = "MI remote",
                             "MI_IZ" = "MI IZ"))

## Mean plot with points
goi <- "Thbd"
thbd_plot <- plot_proteomics_boxplot(norm_table = protein_sub,
                                    protein = goi,
                                    style = "mean") +
  theme(axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) +
  geom_signif(comparisons = list(c("MI_IZ","MI_remote")),
              tip_length = 0, annotation = "0.001", y_position = 16.5) +
  geom_signif(comparisons = list(c("MI_IZ","Control")),
              tip_length = 0, annotation = "0.023", y_position = 15.5) +
  labs(x = "") +
  ylim(yaxis_limits) + 
  labs(y = "") +
  scale_x_discrete(labels=c("Control" = "Control", 
                            "MI_remote" = "MI remote",
                             "MI_IZ" = "MI IZ"))

## Median plot with points
goi <- "Vcam1"
vcam1_plot <- plot_proteomics_boxplot(norm_table = protein_sub,
                                    protein = goi,
                                    style = "mean") +
  theme(axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) +
  geom_signif(comparisons = list(c("MI_IZ","MI_remote")),
              tip_length = 0, annotation = "6.7e-4", y_position = 16.5) +
  geom_signif(comparisons = list(c("MI_IZ","Control")),
              tip_length = 0, annotation = "0.0078", y_position = 15.5) +
  labs(x = "") +
  ylim(yaxis_limits) + 
  labs(y = "") +
  scale_x_discrete(labels=c("Control" = "Control", 
                            "MI_remote" = "MI remote",
                             "MI_IZ" = "MI IZ"))


## Median plot with points
goi <- "Vwf"
vwf_plot <- plot_proteomics_boxplot(norm_table = protein_sub,
                                    protein = goi,
                                    style = "mean") +
  geom_signif(comparisons = list(c("MI_IZ","MI_remote")),
              tip_length = 0, annotation = "0.0057", y_position = 16.5) +
  geom_signif(comparisons = list(c("MI_IZ","Control")),
              tip_length = 0, annotation = "0.0022", y_position = 15.5) +
  theme(axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) +
  labs(x = "") +
  ylim(yaxis_limits) +
  scale_x_discrete(labels=c("Control" = "Control", 
                            "MI_remote" = "MI remote",
                             "MI_IZ" = "MI IZ"))

# save_plot(filename = "./figures/Figure_5.vwf_expression_plot.pdf",
#           plot = vwf_plot,
#           base_asp = 0.5,
#           base_height = 4)


joined_plot <- npr3_plot + thbd_plot +  vcam1_plot + vwf_plot + plot_layout(nrow = 1,axis_titles = "collect") & labs(y = "Normalized protein level")

save_plot(filename = "./plots/Figure_5.expression_plot.pdf",
          plot = joined_plot,
          base_asp = 2.5,
          base_height = 4)
```

# Combine plots for figure

```{r}
# # Combine plots
# comb_plot <- (pca_plot + volc_limma_IZ_remote + path_plot) / (endo_proteomic_corr + wrap_plots(npr3_plot,vwf_plot))
# 
# save_plot(filename = "./figures/Figure_4.proteomics_combined.pdf",
#           plot = comb_plot,
#           base_asp = 2.5,
#           base_height = 12)
```


# Subfigure H - human CITE-seq umap

We first read in the rds object for the human CITE-Seq data from [Amrute et al. 2022](https://www.biorxiv.org/content/10.1101/2022.10.17.512579v1).

```{r}
library(Seurat)
library(SCpubr)

human_citeseq <- readRDS("../public_data/Amrute_et_al/final_global_annotated.rds")
```

We then plot the UMAP of their SCT data to highlight the endocardial cell cluster.

```{r}
DefaultAssay(human_citeseq) <- "SCT"

#Choose endocardial cluster
Idents(human_citeseq) <- "annotation.0.1"
human_endocardium <- subset (human_citeseq, idents = "Endocardium")

#Choose Donor and AMI only
Idents(human_endocardium) <- "HF.etiology"
human_endocardium <- subset (human_endocardium, idents = c("Donor", "AMI"))

#plot VWF expression
plot3 <- VlnPlot (human_endocardium, feature = c("VWF"), cols = c("#008000", "#CD1076"))

#plot Umap embedding using SCpubr package
named_colors <- c("Fibroblast" = "#1f77b4",
                    "B Cells" = "#d62728",
                    "Plasma Cells" = "#ff7f0e",
                    "Endocardium" = "#17becf",
                    "Endothelium" = "#8c564b",
                    "Lymphatics" = "lightgrey",
                    "T_NK Cells" = "#bcbd22",
                    "Myeloid" = "#2ca02c",
                    "Glia" = "#9467bd",
                    "SMC_Pericyte" = "#e377c2", 
                    "Mast Cells" = "darkred")

human_cite_umap <- SCpubr::do_DimPlot(sample = human_citeseq, 
                            label = FALSE, label.box= TRUE, 
                            group.by = "annotation.0.1", 
                            repel = TRUE, 
                            legend.position = "right", plot_cell_borders = TRUE, 
                            plot_density_contour = FALSE,
                            plot.axes = FALSE, raster.dpi = 300, 
                            shuffle = FALSE, 
                            pt.size = 0.4, reduction = "rna.umap",
                            legend.icon.size = 5, 
                            legend.byrow = TRUE, colors.use = named_colors) +
  theme(legend.position = "none")

save_plot(human_cite_umap,
          file = "./plots/Figure4.human_citeseq_umap.png",
          base_height = 3,
          base_asp = 1.3)
```

# Subfigure I - Violin plot for vWF

We want to find out, whether vWF is differentially expressed between Donor and AMI samples in the human endocardial cell cluster. To do so, we will generate pseudobulk data for the endocardial cluster per sample and then compare Donor vs AMI groups. This way, we are basically performing a DE analysis at the pseudobulk level, taking biological replicates into account, while not inflating the test statistics with single-cell measurements.

```{r}
sub_human_cite <- subset(human_endocardium,HF.etiology %in% c("Donor","AMI"))
sub_human_cite$disease_group <- sub_human_cite$HF.etiology

## Quick DE analysis between Donor and AMI
sub_human_cite_pb <- AggregateExpression(sub_human_cite,
                                   return.seurat = T,
                                   group.by = c("sample","disease_group"))
Idents(sub_human_cite_pb) <- "disease_group"

sub_human_cite_pb_de <- FindMarkers(object = sub_human_cite_pb,
                         ident.1 = "Donor",
                         ident.2 = "AMI",
                         test.use = "DESeq2",
                         assay = "RNA",
                         slot = "counts",
                         min.pct = 0.1)

sub_human_cite_pb_de$gene <- rownames(sub_human_cite_pb_de)
sub_human_cite_pb_vwf <- subset(sub_human_cite_pb_de,gene == "VWF")

## We will fetch the p-value for vWF as we are specifically looking at this one gene
pvalue <- sub_human_cite_pb_vwf$p_val
pvalue
```

Finally, let's plot a violin plot for all cells across all samples in the endocardial cluster. First, we will plot a version with only two violins, one for each group.

```{r}
Idents(sub_human_cite) <- sub_human_cite$disease_group
vwf_vlnpot <- SCpubr::do_ViolinPlot(sample = sub_human_cite, 
                      features = "VWF",
                      group.by = "disease_group",
                      line_width = 1,
                      legend.position = "none",
                      legend.title = "",
                      font.size = 25,
                      ylab = "Expression level",
                      xlab = "",
                      colors.use = c("Donor" = "#008000", 
                                     "AMI" = "#CD1076",
                                     "ICM" = "white",
                                     "NICM" = "white"))

vwf_vlnpot <- vwf_vlnpot + theme(plot.margin = margin(t=10, r=10, b=-25, l=10, unit="pt"))

vwf_vlnpot

# save_plot(vwf_vlnpot,
#           file = "./plots/Figure4.human_citeseq_vlnplot.pdf",
#           base_height = 4)
```

As a second visualization option, we will plot a boxplot to show the pseudobulk values that were used in the statistical tests. Here, each dot represents the pseudobulk value for 1 sample for each group (Donor n=6, AMI n=4).

```{r}
Idents(sub_human_cite_pb) <- sub_human_cite_pb$disease_group
sub_human_cite_pb$disease_group <- factor(sub_human_cite_pb$disease_group, levels = c("Donor","AMI"))
vwf_vlnpot_pseudobulk <- VlnPlot(sub_human_cite_pb,
        features = "VWF",
        group.by = "disease_group",
        pt.size = 4,
        cols = c("#008000", "#CD1076"))

save_plot(vwf_vlnpot_pseudobulk,
          file = "./plots/Figure5.human_citeseq_vlnplot.pdf",
          base_height = 4,
          base_asp = 2.25)
```


# Save differential protein expression results for Table 3

```{r}
table3 <- limma_res %>%
  select(-c(label_protein,"P.Value"))

colnames(table3) <- gsub("\\.","_",colnames(table3))
colnames(table3) <- gsub("adj_P_Val","ajusted_pval",colnames(table3))

table3 <- table3 %>%
  select(analysis,logFC,AveExpr,t,pval,ajusted_pval,B,gene,protein_ids) %>%
  arrange(desc(logFC)) %>%
  drop_na()

write.table(table3,
            file = "./output/proteomics/Table3.tsv",
            sep = "\t",
            quote = F,
            row.names = F,
            col.names = TRUE)

```
