---
title: "Molecular Cartography - Seurat analysis"
author: "FloWuenne"
date: "2023-12-06"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

```{r}
library(Seurat)
library(SeuratData)
library(SeuratDisk)
options(Seurat.object.assay.version = "v5")
library(data.table)
library(harmony)
library(here)
library(Nebulosa)
library(tidyverse)
library(GGally)

source("./code/functions.R")
```

# Introduction

In this analysis, we will process the single-cell quantification from nf-Molkart. These tables are basically the output from the regionprops command from scikit and contain properties from the segmentation masks, as well as the count of RNA molecules per gene. We will parse these tables to extract the transcript counts, put these in the expression matrix slots from Seurat and extract the metadata for samples and segmentation masks and put these into the metadata slot of Seurat objects.

# Read data

## Process mcquant tables to seurat object list

```{r}
final_samples <- c("sample_control_r1_s1","sample_control_r2_s1",
                   "sample_4h_r1_s1","sample_4h_r2_s2",
                   "sample_2d_r1_s1","sample_2d_r2_s1",
                   "sample_4d_r1_s1","sample_4d_r2_s1")

seurat_objects <- list()

## SCTransform based clustering across all samples
sample_dir <- "../results/nf-core_molkart/spot2cell"
seurat_list <- list()
segmethod_samples <- list.files(sample_dir)
for(sample in segmethod_samples){
    sample_ID <- strsplit(sample,"\\.")[[1]][1]
    sample_ID <- strsplit(sample_ID,split="_")[[1]][-1]
    sample_ID <- paste(sample_ID[1:length(sample_ID) -1],collapse = "_")
    if(endsWith(sample,"_cellpose.csv") & sample_ID %in% final_samples){
      print(sample_ID)
      sample_quant <- fread(paste(sample_dir,sample,sep="/"))
      seurat_object <- create_seurat_sctransform_mcquant(sample_quant,sample_ID)
      seurat_list[[sample_ID]] <- seurat_object
  }
}

resolve_object <- merge(seurat_list[[1]], y = c(seurat_list[-1]), project = "Molecular_Cartography")
```

# Process data

## Plot distribution of transcript counts and mask properties

```{r}
## Set order of timepoints
resolve_object@meta.data$timepoint <- factor(resolve_object@meta.data$timepoint,
                               levels = c("control","4h","2d","4d"))


## Show distribution of QC measures before filtering outliers
VlnPlot(resolve_object, features = c("nCount_RNA","nFeature_RNA","Area","Eccentricity","Solidity"),
        group.by = "timepoint", pt.size = 0, cols = time_palette)
```

## Filter outlier cells

```{r}
resolve_object <- subset(resolve_object, subset = nCount_RNA < 1000 & Eccentricity > 0.4 & Solidity > 0.7 & Area < 30000 & nCount_RNA > 10 & nFeature_RNA <= 40)
```

```{r}
## Show distribution of QC measures after filtering outliers
VlnPlot(resolve_object, features = c("nCount_RNA","nFeature_RNA","Area","Eccentricity","Solidity"),
        group.by = "timepoint", pt.size = 0, cols = time_palette)
```

## Integrate samples using Harmony

```{r}
## Normalize and scale data using sctransform v2
resolve_object <- SCTransform(resolve_object, assay = "RNA")
```


```{r}
## Perform PCA
npcs_to_use <- 20
resolve_object <- RunPCA(resolve_object, npcs = npcs_to_use, verbose = FALSE, approx=FALSE)
```

```{r}
## Run Harmony to integrate datasets across batches and time
# harmony_object <- RunHarmony(resolve_object, 
#                              group.by.vars = c("timepoint","sample_ID","slide"), 
#                              plot_convergence = FALSE,
#                              assay = "SCT",
#                              dims.use = 1:npcs_to_use)

harmony_object <- RunHarmony(resolve_object,
                             group.by.vars = c("timepoint"),
                             plot_convergence = FALSE,
                             assay = "SCT",
                             dims.use = 1:npcs_to_use)
```


```{r}
harmony_object <- RunUMAP(harmony_object, reduction = "harmony", dims = 1:npcs_to_use)
harmony_object <- FindNeighbors(harmony_object, reduction = "harmony", dims = 1:npcs_to_use) %>% FindClusters(resolution = 0.3)
DimPlot(harmony_object, raster = FALSE)
```

## Plot Harmony integration results

```{r}
## Plot UMAP with cells labeled by sampled time point and by cluster assigned using nearest-neighbor analysis
p1 <- DimPlot(harmony_object, reduction = "umap", group.by = "timepoint",raster=FALSE, 
              cols = time_palette, alpha = 0.5)
p2 <- DimPlot(harmony_object, reduction = "umap", label = TRUE, repel = TRUE,raster=FALSE)
p1+p2
```

```{r}
## Plot UMAP next to a barplot
meta_time <- harmony_object@meta.data %>%
  group_by(seurat_clusters, timepoint) %>%
  tally() %>%
  mutate("frac" = n / sum(n))

time_bar <- ggplot(meta_time,aes(seurat_clusters,frac, fill = timepoint)) +
  geom_bar(stat = "identity", position = "stack") +
  coord_flip() +
  scale_fill_manual(values = time_palette)

combined_time_plot <- p2 + time_bar & 
  theme(plot.background = element_rect(fill = 'white'),
        panel.background = element_rect(fill = 'white'))

save_plot(combined_time_plot,
          file = "./plots/molkart.umap_time.png",
          base_height = 10)
```

```{r}
## Plot Nebulosa plots for QC measures to highlight cell clusters with high number of transcripts
plot_density(harmony_object, features= c("Area","nCount_RNA","nFeature_RNA"))
```

```{r}
## Plot the number of cells per cluster and color by cluster ID
cell_numbers <- harmony_object@meta.data %>%
  group_by(seurat_clusters) %>%
  tally()

ggplot(cell_numbers,aes(seurat_clusters,n)) +
  geom_bar(stat = "identity",aes(fill = seurat_clusters)) +
  theme(legend.position = "None")
```

## Identify marker genes for cell clusters

```{r}
harmony_object <- PrepSCTFindMarkers(harmony_object, assay = "SCT", verbose = TRUE)
harmony_markers <- FindAllMarkers(harmony_object, logfc.threshold = 0.5, only.pos = TRUE)
```

## Transfer labels from snRNA-seq (Calcagno et al. 2020)

```{r}
calcagno_et_al <- LoadH5Seurat("../public_data/Calcagno_et_al_NatCardioVasc_2022/reprocessed_data/Calcagno2022_int_logNorm_annot.h5Seurat")
```

```{r}
calcagno_et_al <- SCTransform(calcagno_et_al, assay = "RNA")
```

```{r}
anchors <- FindTransferAnchors(reference = calcagno_et_al, query = harmony_object, normalization.method = "SCT", npcs = 50, recompute.residuals = FALSE)
predictions.assay_lvl1 <- TransferData(anchorset = anchors, refdata = calcagno_et_al$level_1, prediction.assay = FALSE, weight.reduction = "pcaproject", dims = NULL)
predictions.assay_lvl2 <- TransferData(anchorset = anchors, refdata = calcagno_et_al$level_2, prediction.assay = FALSE, weight.reduction = "pcaproject", dims = NULL)
predictions.assay_lvl3 <- TransferData(anchorset = anchors, refdata = calcagno_et_al$level_3, prediction.assay = FALSE, weight.reduction = "pcaproject", dims = NULL)
predictions.orig_label <- TransferData(anchorset = anchors, refdata = calcagno_et_al$orig_label, prediction.assay = FALSE, weight.reduction = "pcaproject", dims = NULL)
```


```{r}
## Add labels to Harmony object and plot Umap
harmony_object@meta.data$predicted_ct_lvl1 <- predictions.assay_lvl1$predicted.id
harmony_object@meta.data$predicted_ct_lvl2 <- predictions.assay_lvl2$predicted.id
harmony_object@meta.data$predicted_ct_lvl3 <- predictions.assay_lvl3$predicted.id
harmony_object@meta.data$orig_label_calcagno <- predictions.orig_label$predicted.id

DimPlot(harmony_object, reduction = "umap", group.by = "predicted_ct_lvl1",raster=FALSE, label = TRUE)
DimPlot(harmony_object, reduction = "umap", group.by = "predicted_ct_lvl2",raster=FALSE, label = TRUE)
DimPlot(harmony_object, reduction = "umap", group.by = "predicted_ct_lvl3",raster=FALSE, label = TRUE)
DimPlot(harmony_object, reduction = "umap", group.by = "orig_label_calcagno",raster=FALSE, label = TRUE)
```

## Assign cell-type names to clusters

```{r}
## Annotate main cell-types and states
## LVL1
seurat_clusters <- harmony_object@active.ident

## Cluster level 1
new.cluster.ids <- c("Endothelial_cells",  #0
                     "Cardiomyocytes",  #1
                     "Endothelial_cells", #2
                     "Cardiomyocytes", #3
                     "Cardiac_fibroblasts", #4
                     "Cardiomyocytes", #5
                     "Cardiac_fibroblasts", #6
                     "Cardiomyocytes", #7
                     "Immune_cells", #8
                     "Endothelial_cells", #9
                     "Immune_cells", #10
                     "Pericytes" , #11
                     "Cardiomyocytes", #12
                     "Immune_cells", #13
                     "Smooth_muscle_cells", #14
                     "Cardiomyocytes", #15
                     "Endothelial_cells", #16
                     "Endothelial_cells",#17
                     "Immune_cells", #18
                     "Endothelial_cells" #19
                     )

names(new.cluster.ids) <- levels(harmony_object)
harmony_object <- RenameIdents(harmony_object, new.cluster.ids)
harmony_object@meta.data$anno_cell_type_lvl1 <- harmony_object@active.ident

## Reset cluster to seurat ID
Idents(object = harmony_object) <- "seurat_clusters"
```


```{r}
## Annotate cell clusters with more detaile
## Lvl2

seurat_clusters <- harmony_object@active.ident

## Cluster level 2
new.cluster.ids <- c("Vascular_endothelium",  #0 - Kdr
                     "Cardiomyocytes",  #1 high nCount_rna
                     "Vascular_endothelium", #2 - Aqp1
                     "Cardiomyocytes", #3 Ctnnb1
                     "Cardiac_fibroblasts", #4 Col1a1
                     "Cardiomyocytes_Nppa+", #5 Nppa
                     "Cardiac_fibroblasts", #6 Dcn, Pdgfra
                     "Cardiomyocytes_Nppb+", #7 Nppb
                     "Myeloid_cells", #8 Cs1f1r
                     "EndMA+", #9 Fn1, Mmp14
                     "Myeloid_cells", #10 
                     "Pericytes" , #11
                     "Cardiomyocytes_Nppa+", #12
                     "Myeloid_cells", #13
                     "Smooth_muscle_cells", #14
                     "Cardiomyocytes", #15
                     "Vascular_endothelium", #16
                     "Endocardial_cell",#17
                     "Lymphoid cells", #18
                     "Vascular_endothelium" #19
                     )

names(new.cluster.ids) <- levels(harmony_object)
harmony_object <- RenameIdents(harmony_object, new.cluster.ids)
harmony_object@meta.data$anno_cell_type_lv2 <- harmony_object@active.ident

## Reset cluster to seurat ID
Idents(object = harmony_object) <- "seurat_clusters"
```

```{r}
## Annotate cell clusters with more detaile
## Lvl3

seurat_clusters <- harmony_object@active.ident

## Cluster level 3
new.cluster.ids <- c("Vascular_endothelium_Kdr",  #0 - Kdr
                     "Cardiomyocytes_high_nCountRNA",  #1 high nCount_rna
                     "Vascular_endothelium_Aqp1", #2 - Aqp1
                     "Cardiomyocytes_Ctnnb1", #3 Ctnnb1
                     "Cardiac_fibroblasts_Col1a1", #4 Col1a1
                     "Cardiomyocytes_Nppa+", #5 Nppa
                     "Cardiac_fibroblasts_Pdgfra", #6 Dcn, Pdgfra
                     "Cardiomyocytes_Nppb+", #7 Nppb
                     "Monocytes_Macrophages_Trem2", #8 Cs1f1r
                     "EndMA+", #9 Fn1, Mmp14
                     "Monocytes_Macrophages_H2-Eb1", #10 
                     "Pericytes_Colec11" , #11
                     "Cardiomyocytes_Nppa+", #12
                     "Monocytes_Macrophages_Trem2", #13
                     "Smooth_muscle_cells_Myh11", #14
                     "Cardiomyocytes_Vegfa", #15
                     "Vascular_endothelium", #16
                     "Endocardial_cells_Npr3",#17
                     "B-cells_Ighm", #18
                     "Endothelial_cells" #19
                     )

names(new.cluster.ids) <- levels(harmony_object)
harmony_object <- RenameIdents(harmony_object, new.cluster.ids)
harmony_object@meta.data$anno_cell_type_lvl3 <- harmony_object@active.ident

## Reset cluster to seurat ID
Idents(object = harmony_object) <- "seurat_clusters"
```

```{r}
## Additionally, we will label all cells that express Ccr2 over a threshold
ccr2_exp <- hist(harmony_object@assays$SCT@scale.data["Ccr2",])
ccr2_cells <- harmony_object@assays$SCT@data["Ccr2",] > 0
harmony_object@meta.data$ccr2_pos <- if_else(harmony_object@assays$SCT@data["Ccr2",] > 0,"yes","no")
```

```{r}
Idents(object = harmony_object) <- "anno_cell_type_lv2"
harmony_object <- PrepSCTFindMarkers(harmony_object, assay = "SCT", verbose = TRUE)
harmony_markers <- FindAllMarkers(harmony_object, logfc.threshold = 0.5, only.pos = TRUE)
```


```{r}
library(viridis)
genes <- c("Ighm","Npr3","Acta2","Colec11","Fn1","Lyz2","Clu","Nppa","Dcn","Ryr2","Aqp1")
DotPlot(harmony_object, group.by = "anno_cell_type_lv2",
        features = c("Ighm","Npr3","Acta2","Colec11","Fn1","Lyz2","Clu","Nppa","Dcn","Ryr2","Aqp1")) +
  geom_point(aes(size=pct.exp), shape = 21, colour="black", stroke=0.5) +
  scale_colour_viridis(option="magma", direction = -1) +
  guides(size=guide_legend(override.aes=list(shape=21, colour="black", fill="white")))
```


```{r}
# library(pheatmap)
# ## Instead of a dotplot, let's look at a pseudbulk heatmap
# pseudobulk <- AverageExpression(harmony_object, assay = "SCT",group.by = "anno_cell_type_lv2",slot = "data")
# pheatmap(pseudobulk$SCT[genes,])
```


# Save object as seurat RDS

```{r}
saveRDS(harmony_object,
        file = "./output/molkart/molkart.harmony_seurat_object.rds")
```


```{r}
## If the object has already been computed
harmony_object <- readRDS(file = "./output/mol_cart/molkart.harmony_seurat_object.rds")
```

# Save data formatted for input to spatial anndata object

```{r}
metadata <- harmony_object@meta.data
coordinates <- metadata %>% select(X_centroid,Y_centroid)
counts <- t(as.matrix(harmony_object@assays$SCT@counts))

## Save coordinates
write.table(coordinates,
            "./output/mol_cart/harmony_object.coordinates_anndata.tsv",
            sep="\t",
            col.names = FALSE,
            row.names = FALSE,
            quote = FALSE)

## Save counts
write.table(counts,
            "./output/mol_cart/harmony_object.counts_anndata.tsv",
            sep="\t",
            col.names = TRUE,
            row.names = FALSE,
            quote = FALSE)

## Annotation
write.table(metadata,
            "./output/mol_cart/harmony_object.metadata_anndata.tsv",
            sep="\t",
            col.names = TRUE,
            row.names = FALSE,
            quote = FALSE)
```

# Generate Misty input

```{r}
misty_in <- harmony_object@meta.data %>%
  dplyr::select(sample_ID,timepoint,replicate,X_centroid,Y_centroid,Area,anno_cell_type_lvl1,anno_cell_type_lv2,anno_cell_type_lvl3,ccr2_pos,nCount_RNA) 
misty_in$cell_ID <- rownames(misty_in)
```

```{r}
expression_plot_list <- list()

samples <- c("sample_control_r1_s1","sample_4h_r1_s1",
             "sample_2d_r1_s1","sample_4d_r1_s1",
             "sample_control_r2_s1","sample_4h_r2_s2",
             "sample_2d_r2_s1","sample_4d_r2_s1")

for(this_sample in samples){
  pt_size <- 0.2
  cluster_of_int <- c(16,19)
  sample_object <- subset(misty_in,sample_ID == this_sample)
  
  highlight_plot <- ggplot(sample_object,aes(X_centroid,Y_centroid)) +
    geom_point(aes(color = anno_cell_type_lv2),size = pt_size) +
    theme_classic() +
    labs(x = "Spatial 1",
         y = "Spatial 2",
         title = this_sample) +
    theme(axis.ticks = element_blank(),
            axis.text = element_blank(),
            legend.position = "right") +
    scale_color_brewer(palette = "Set3")
  
      expression_plot_list[[this_sample]] <- highlight_plot
  
}

wrap_plots(expression_plot_list, nrow = 2, ncol = 4)  + plot_layout(guides = 'collect')
```

```{r}
write.table(misty_in,
            file = "./output/mol_cart/molcart.misty_celltype_table.tsv",
            sep="\t",
            col.names = TRUE,
            row.names = FALSE,
            quote = FALSE)
```

# Plot cell-type on spatial plot

```{r}
expression_plot_list <- list()

samples <- c("sample_control_r1_s1","sample_4h_r1_s1",
             "sample_2d_r1_s1","sample_4d_r1_s1",
             "sample_control_r2_s1","sample_4h_r2_s2",
             "sample_2d_r2_s1","sample_4d_r2_s1")

for(cluster_of_int in unique(harmony_object$anno_cell_type_lv2)){
  print(cluster_of_int)
  for(this_sample in samples){
    pt_size <- 0.6
    sample_object <- subset(harmony_object,sample_ID == this_sample)
    meta <- sample_object@meta.data
    
    highlight_plot <- ggplot(meta,aes(Y_centroid,X_centroid)) +
      geom_point(data = subset(meta,!seurat_clusters %in% cluster_of_int),color = "darkgrey", size = pt_size) +
      # geom_point(data = subset(meta,seurat_clusters %in% cluster_of_int),aes(color = seurat_clusters), size = pt_size) +
      geom_point(data = subset(meta,anno_cell_type_lv2 == cluster_of_int),color = "purple", size = 1) +
      theme_classic() +
      labs(x = "Spatial 1",
           y = "Spatial 2") +
      theme(axis.ticks = element_blank(),
            axis.text = element_blank(),
            legend.position = "right")
    
    expression_plot_list[[this_sample]] <- highlight_plot
    }
  
  time_plot <- wrap_plots(expression_plot_list, nrow = 2, ncol = 4)  + plot_layout(guides = 'collect') + plot_annotation(cluster_of_int,theme=theme(plot.title=element_text(hjust=0.5)))
  filename <- paste("./output/molkart_cell_types/",cluster_of_int,".cell_type_distributions.png",sep = "")
  save_plot(time_plot,
            filename = filename,
            base_height = 6)
}
```

```{r}
test <-as.data.frame(harmony_object@assays$SCT@counts)
```

```{r}
colnames(test) <- 1:ncol(test)
```

