---
title: "Deep Visual Proteomics - snRNA-seq comparison"
author: "Florian Wuennemann"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
library(Seurat)
library(here)
library(Libra)
library(Nebulosa)
library(data.table)
library(ggrepel)
library(plotly)
library(SeuratDisk)

source("./code/functions.R")
```

## Introduction

# Load data

We reprocessed the data from Calcagno et al, using the original cell-type annotations from the paper and will load this dataset as a seurat object.


```{r}
## Load reprocessed Calcagno et al seurat object
calcagno_et_al <- LoadH5Seurat("./data/Calcagno2022_int_logNorm_annot.h5Seurat")
```

```{r}
## Get only control cells for marker calculation
calcagno_et_al_ds <- subset(calcagno_et_al,time == "D0")
calcagno_et_al_d1 <- subset(calcagno_et_al,time == "D1")
```


Let's check the UMAP embedding from our reprocessed object.

```{r}
## UMAP plot
DimPlot(calcagno_et_al,label = TRUE) + theme(legend.position = "none")
```

Next, let's quickly verify, that Endocardial cells are expressing the proper markers before we compare them to our proteomic data.

```{r}
plot_density(calcagno_et_al, features = "Npr3")
VlnPlot(calcagno_et_al, features = "Npr3")
```

Expression of the endocardial specific marker Npr3 in this dataset fits with the original authors annotation, suggesting that we can use these endocardial single-cell signature to identify endocardial specific genes in our proteomics data.


# Analyze cell-type specific proteins in proteomic data

We will use the snRNAseq data to identify proteins likely differentially expressed in endocardial cells. For this, we will first identify genes specifically expressed in endocardial cells.

## Correlate pseudobulk snRNA-seq expression in endocardial cells with proteomic measurements

Let's load the proteomic data now: 

```{r}
limma_res <- fread("./output/proteomics/proteomics.limma.full_statistics.tsv")

## Extract statistics for different contrasts
miiz_vs_control_signature <- subset(limma_res,analysis == "MI_IZ_vs_control")
miiz_vs_remote_signature <- subset(limma_res,analysis == "MI_IZ_vs_MI_remote")

## Load the normalized protein matrix as well
protein_mat <- fread(file = "./output/proteomics/proteomics.vsn_norm_proteins.tsv")
protein_mat_avg <- protein_mat %>%
  mutate(avg_control=rowMeans(.[ , c("control_r1","control_r2","control_r3")], na.rm=TRUE)) %>%
  mutate(avg_MI_IZ=rowMeans(.[ , c("MI_IZ_r1","MI_IZ_r2","MI_IZ_r3","MI_IZ_r4")], na.rm=TRUE)) %>%
  mutate(avg_MI_remote=rowMeans(.[ , c("MI_remote_r1","MI_remote_r2","MI_remote_r3","MI_remote_r4")], na.rm=TRUE)) %>%
  dplyr::select(gene,avg_control,avg_MI_IZ,avg_MI_remote)
```


```{r}
## Calculate pseudobulk expression profiles for endocardial cells
endocard_seurat <- subset(calcagno_et_al, level_2 == "Endocardial")
sn_endo_bulk <- AverageExpression(endocard_seurat, group.by = c("time"),slot= "data")
sn_endo_bulk_df <- as.data.frame(sn_endo_bulk$RNA)
sn_endo_bulk_df$gene <- rownames(sn_endo_bulk_df)
```


```{r}
## Merge average protein expression values with average RNA expression
rna_protein_avg <- left_join(protein_mat_avg,sn_endo_bulk_df, by = "gene") %>%
  drop_na()

corrplot_rna_protein <- ggplot(rna_protein_avg,aes(avg_control,D0, label = gene)) +
  geom_point() +
  geom_point(data = subset(rna_protein_avg, gene == "Vwf"),color = "red", size =3) +
  labs(x = "Average protein expression (Control)",
       y = "Average snRNA-seq expression Control)")
corrplot_rna_protein
```


## Calculate differentially expressed genes in endocardial cells in snRNA-seq


```{r}
calcagno_et_al$cell_type_time <- paste(calcagno_et_al$level_2, calcagno_et_al$time,
    sep = "_")
Idents(calcagno_et_al) <- "cell_type_time"

endocard_de <- FindMarkers(calcagno_et_al, 
                           ident.1 = "Endocardial_D1", 
                           ident.2 = "Endocardial_D0",
                           min.diff.pct = 0.1,
                           logfc.threshold = 0,
                           verbose = FALSE)

colnames(endocard_de) <- gsub("\\.","_",colnames(endocard_de))
endocard_de <- endocard_de %>% 
  mutate("gene" = rownames(endocard_de)) %>%
  mutate("pct_ratio" = pct_2 /pct_1,
         "pct_diff" = pct_2 -pct_1) %>%
  arrange(desc(avg_log2FC))
```

## Get endocardial specific genes

To get an estimate of which genes are most specifically expressed in endocardialc ells in the snRNA-seq data, we will use the FindMarkers function from Seurat to get log-fold changes and p-values for the comparison of endocardial cells from Sham controls to all other cells.

```{r}
endo_marker <- FindMarkers(calcagno_et_al,ident.1 = "Endocardial_D0",
                           only.pos = TRUE)
```

```{r}
endo_marker$gene <- rownames(endo_marker)
endo_marker <- endo_marker %>%
  mutate("pct_diff" = pct.1 - pct.2) %>% # Only 
  mutate("pct_ratio" = pct.1 / pct.2) %>%
  subset(pct.2 < 0.1)                         
```

## Compare endocardial specific genes with differentially expressed proteins (DEPs)

```{r}
merged_protein_rna <- left_join(endo_marker,miiz_vs_remote_signature, by = "gene")

merged_protein_rna <- merged_protein_rna %>%
  mutate("label_gene" = if_else(gene %in% c("Vwf","Npr3"),gene,""))

endo_proteomic_corr <- ggplot(merged_protein_rna,aes(avg_log2FC,logFC,
                              label = label_gene)) +
  geom_point(data =subset(merged_protein_rna,gene != "Vwf"), size =3, fill = "darkgrey", pch = 21) +
  geom_point(data = subset(merged_protein_rna,gene == "Vwf"),size = 4, fill = "red", pch = 21) +
  geom_point(data = subset(merged_protein_rna,gene == "Npr3"),size = 4, fill = "purple", pch = 21) +
  geom_label_repel() +
  labs(x = "Specificity for endocardial cells (snRNA-seq)",
       y = "Log-fold change MI_IZ vs control(proteomics)")

endo_proteomic_corr
```

```{r}
write.table(merged_protein_rna,
            file = "./output/proteomics/proteomics.snRNAseq_comp.tsv",
            sep = "\t",
            col.names = TRUE,
            row.names = FALSE,
            quote = FALSE)
```
